<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.15.2 by Michael Rose
  Copyright 2013-2019 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>June SciComp Updates - Fred Hutch Biomedical Data Science Wiki</title>
<meta name="description" content="Latest R:  R/3.5.0-foss-2016b-fh1Latest Python:  Python/3.6.5-foss-2016b-fh3">



<meta property="og:type" content="article">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Fred Hutch Biomedical Data Science Wiki">
<meta property="og:title" content="June SciComp Updates">
<meta property="og:url" content="http://localhost:4000/scicompannounce/2018-06-01-JuneComputingUpdates/">


  <meta property="og:description" content="Latest R:  R/3.5.0-foss-2016b-fh1Latest Python:  Python/3.6.5-foss-2016b-fh3">







  <meta property="article:published_time" content="2018-06-01T00:00:00-07:00">





  

  


<link rel="canonical" href="http://localhost:4000/scicompannounce/2018-06-01-JuneComputingUpdates/">







  <script type="application/ld+json">
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000",
      "sameAs": null
    }
  </script>







<!-- end _includes/seo.html -->


<link href="index.html" type="application/atom+xml" rel="alternate" title="Fred Hutch Biomedical Data Science Wiki Feed">

<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">

<!--[if IE ]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->



    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
              <a href="http://localhost:4000"><img src="/images/Biomedical-Data-Science-Wiki-logo.png" alt="FH biomedical data science wiki icon" width="600" align="left|middle" hspace="0"></a>
        <!--<a class="site-title" href="/">Fred Hutch Biomedical Data Science Wiki</a>-->

        <ul class="visible-links">
          
          
          <li class="masthead__menu-item">
            <a href="/generation/gen_index/" >Data Generation</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/bioinformatics/inf_index/" >Bioinformatics</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/computing/comp_index/" >Computing</a>
          </li>
          
        </ul>
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z"
              transform="translate(-.01)"></path>
          </svg>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">
  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Computing</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/computing/comp_index/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/compdemos/" class="">Resource Library/HOWTOs</a></li>
          
            
            

            
            

            <li><a href="/scicompannounce/" class="">Announcements</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Access and Credentials</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/computing/access_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/computing/access_credentials/" class="">Credentials</a></li>
          
            
            

            
            

            <li><a href="/computing/access_methods/" class="">Methods</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Data Storage</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/computing/store_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/computing/store_databases/" class="">Database Systems</a></li>
          
            
            

            
            

            <li><a href="/computing/store_posix/" class="">File Storage Systems</a></li>
          
            
            

            
            

            <li><a href="/computing/store_objectstore/" class="">Object Storage Systems</a></li>
          
            
            

            
            

            <li><a href="/computing/store_scratch/" class="">Temporary (Scratch) Storage</a></li>
          
            
            

            
            

            <li><a href="/computing/store_collaboration/" class="">Collaborative Storage Systems</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Large Scale Compute</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/computing/cluster_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/computing/linux_linux101/" class="">Linux Operating System</a></li>
          
            
            

            
            

            <li><a href="/computing/resource_overview/" class="">Technologies</a></li>
          
            
            

            
            

            <li><a href="/computing/cluster_software/" class="">Scientific Software</a></li>
          
            
            

            
            

            <li><a href="/computing/cluster_usingSlurm/" class="">Job Management</a></li>
          
            
            

            
            

            <li><a href="/computing/cluster_cloudCompute/" class="">Cloud Computing</a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>
    
  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="June SciComp Updates">
    <meta itemprop="description" content="Latest R:  R/3.5.0-foss-2016b-fh1Latest Python:  Python/3.6.5-foss-2016b-fh3">
    <meta itemprop="datePublished" content="June 01, 2018">
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">June SciComp Updates
</h1>
          
          
        </header>
      


      <section class="page__content" itemprop="text">
        
        <p>Latest R:  R/3.5.0-foss-2016b-fh1
Latest Python:  Python/3.6.5-foss-2016b-fh3</p>

<p>For a complete list of modules built with R and Python visit: https://fredhutch.github.io/easybuild-life-sciences/</p>

<p>TeXlive Has been updated to version 2018.05.31.  This new version uses libraries that are compatible with our newest versions of R-3.4.2, 3.4.3 and 3.5.0.  If you have been getting errors with the pdflatex command please try this new build of texlive.
Module name: texlive/20180531-foss-2016b</p>

<p>Python-3.6.0 fh3 - 155 Updated packages, 25 new packages. User requested packages:
https://github.com/wdecoster/nanostat
https://github.com/rrwick/Porechop
https://github.com/wdecoster/NanoPlot</p>

<p>Nanopolish - Software package for signal-level analysis of Oxford Nanopore sequencing data. Nanopolish can calculate an improved consensus sequence for a draft genome assembly, detect base modifications, call SNPs and indels with respect to a reference genome and more.
Module name: nanopolish/0.7.1-foss-2016b</p>

<p>Clairvoyante - A deep neural network based variant caller. Requires DISPLAY to be set. Only works with Python2. Module name: Clairvoyante/0.1</p>

<p>Assembly Stats Get assembly statistics from FASTA and FASTQ files.
Module name: assembly-stats/1.0.1-foss-2016b</p>

<p>Minimap2 is a versatile sequence alignment program that aligns DNA or mRNA sequences against a large reference database. Typical use cases include: (1) mapping PacBio or Oxford Nanopore genomic reads to the human genome; (2) finding overlaps between long reads with error rate up to ~15%; (3) splice-aware alignment of PacBio Iso-Seq or Nanopore cDNA or Direct RNA reads against a reference genome; (4) aligning Illumina single- or paired-end reads; (5) assembly-to-assembly alignment; (6) full-genome alignment between two closely related species with divergence below ~15%.
Module Name: minimap2/2.10-foss-2016b</p>

<p>IUPred Intrinsically unstructured/disordered proteins have no single well-defined tertiary structure in their native, functional state. Our server recognizes such regions from the amino acid sequence based on the estimated pairwise energy content. The underlying assumption is that globular proteins are composed of amino acids which have the potential to form a large number of favorable interactions, whereas intrinsically unstructured proteins (IUPs) adopt no stable structure because their amino acid composition does not allow sufficient favorable interactions to form.                      <br />
Modulename iupred/1.0-GCC-5.4.0-2.26</p>

<p>RAxML Randomized Axelerated Maximum Likelihood. Is a program for sequential and parallel Maximum Likelihood based inference of large phylogenetic trees. It can also be used for post analyses of sets of phylogenetic trees, analyses of alignments and, evolutionary placemen of short reads.
Modulename: RAxML/8.2.11-foss-2016b-hybrid-avx2</p>

<p>Tesseract is an OCR engine.  Tesseract was originally developed at Hewlett-Packard Laboratories Bristol and at Hewlett-Packard Co, Greeley Colorado between 1985 and 1994, with some more changes made in 1996 to port to Windows, and some C++izing in 1998. In 2005 Tesseract was open sourced by HP. Since 2006 it is developed by Google.
Modulename: tesseract/4.0.0-beta.1-foss-2016b</p>

<p>Pindel - Pindel can detect breakpoints of large deletions, medium sized insertions, inversions, tandem duplications and other structural variants at single-based resolution from next-gen sequence data. It uses a pattern growth approach to identify the breakpoints of these variants from paired-end short reads.
Module name: Pindel/0.2.5b8-foss-2016b</p>

<p>R 3.5.0 The latest and largest R yet built. Over 800 packages. Updated with packages from BioConductor 3.7.
Module Name: R/3.5.0-foss-2016b-fh1</p>

<p>Python 2.7.15 268 Packages, 187 Updated Packages
Module Name: Python/2.7.15-foss-2016b-fh1</p>

<p>Leptonica Graphic libraries for image processing and image analysis applications.
Module name: leptonica/1.75.3-foss-2016b</p>

<p>RGEOS  R- Interface to Geometry Engine - Open Source (‘GEOS’) using the C ‘API’ for topology operations on geometries. The ‘GEOS’ library is external to the package, and, when installing the package from source, must be correctly installed first. Windows and Mac Intel OS X binaries are provided on ‘CRAN’. Based on package GEOS-3.6.2 and R-3.5.0
Moduel Name: rgeos/0.3-26-foss-2016b-R-3.5.0</p>

<p>awscli/1.15.16 updated. ASW CLI changes almost weekly.  We are constantly updating this one package.</p>

<p>Steel Bank Common Lisp (SBCL) is a high-performance Common Lisp compiler. It is open source / free software, with a permissive license. In addition to the compiler and runtime system for ANSI Common Lisp, it provides an interactive environment including a debugger, a statistical profiler, a code coverage tool, and many other extensions.
Module Name: sbcl/1.4.6</p>

<p>Sniffles is a structural variation caller using third generation sequencing (PacBio or Oxford Nanopore). It detects all types of SVs (10bp+) using evidence from split-read alignments, high-mismatch regions, and coverage analysis. Please note the current version of Sniffles requires sorted output from BWA-MEM (use -M and -x parameter) or NGMLR with the optional SAM attributes enabled!
Module Name: Sniffles/1.0.8-foss-2016b</p>

<p>NanoSV NanoSV is a software package that can be used to identify structural genomic variations in long-read sequencing data, such as data produced by Oxford Nanopore Technologies’ MinION, GridION or PromethION instruments, or Pacific Biosciences sequencers
Module Name: NanoSV/1.1.2-foss-2016b</p>

<p>SAMtools, BCFtools, HTSlib Samtools is a suite of programs for interacting with high-throughput sequencing data.  All three have been updated to version 1.8.</p>

<p>Python 3.6.5 111 Updates from 3.6.5-fh2.  8 New packages. 460 total packages
Module Name: Python/3.6.5-foss-2016b-fh1</p>

        
      </section>

      <footer class="page__meta">
        
        


        
          <p class="page__date"><strong><i class="fas fa-fw fa-calendar-alt" aria-hidden="true"></i> Updated:</strong> <time datetime="2018-06-01T00:00:00-07:00">June 01, 2018</time></p>
        
      </footer>

      

      
  <nav class="pagination">
    
      <a href="/scicompannounce/2018-05-26-pythonupdate/" class="pagination--pager" title="Python 3.6.5 Software Release
">Previous</a>
    
    
      <a href="/scicompannounce/2019-01-22-hpc-systems-upgrade-2019/" class="pagination--pager" title="High Performance Computing Systems Upgrade 2019 Announcement
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form onsubmit="return googleCustomSearchExecute();" id="cse-search-box-form-id">
    <input type="text" id="cse-search-input-box-id" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
    </form>
    <div id="results" class="results">
        <gcse:searchresults-only></gcse:searchresults-only>    
    </div></div>

      </div>
    

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
          <li><a href="/contributors/"><i class="fas fa-user-edit"></i> See our Contributors</a></li>

          <li><a href="https://github.com/FredHutch/wiki"><i class="fab fa-fw fa-github"></i> See our GitHub Repo</a></li>

          <li><a href="https://fhbig.slack.com/"><i class="fab fa-fw fa-slack"> </i> Join the Conversation on Slack</a></li>

          <li><a href="https://fredhutch.github.io/FHBig/"><img src="/images/FHBig.png" alt="fhbig" width="40" align="left|middle" hspace="0"> Find Upcoming Events</a></li>

          <li><a href="https://www.fredhutch.io/"><img src="/images/fhio.png" alt="fredhutchio" width="20" align="left|middle" hspace="0"> Find Training</a></li>


  </ul>
</div>

<div class="page__footer-copyright">&copy; 2019 Fred Hutch Biomedical Data Science Wiki. Powered by <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>
  <script defer src="https://use.fontawesome.com/releases/v5.7.1/js/all.js" integrity="sha384-eVEQC9zshBn0rFj4+TU78eNA19HMNigMviK/PU/FFjLXqa/GKPgX58rvt5Z8PLs7" crossorigin="anonymous"></script>


<script>
  (function () {
    var cx = '018020560330782233421:waz3ch9w_wg';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();

  function googleCustomSearchExecute() {
    var input = document.getElementById('cse-search-input-box-id');
    var element = google.search.cse.element.getElement('searchresults-only0');
    if (input.value == '') {
      element.clearAllResults();
    } else {
      element.execute(input.value);
    }
    return false;
  }

  
</script>





  </body>
</html>
