---
title: "Python Modules Available"
layout: collection
classes: wide
permalink: /pythonModules/
collection: pythonModules # collection name
entries_layout: list # list (default),
show_excerpts:  true #(default), false
sort_by: date # date (default), title
sort_order:  reverse # (default), reverse, forward
sidebar:
  nav: "scicomputing"
toc: true
toc_label: "On This Page"
---

The Fred Hutch python modules have the suffix `-fh1`. The `-fh1` module is specific to the Hutch, and contains libraries that have been requested by Hutch users.

## Available Python Bundles
<!---DO NOT EDIT BELOW HERE, content comes from easybuild-life-science/docs/python-bundles.md--->

### Biopython/1.74-foss-2016b-Python-3.7.4
Biopython is a set of freely available tools for biological
 computation written in Python by an international team of developers. It is
 a distributed collaborative effort to develop Python libraries and
 applications which address the needs of current and future work in
 bioinformatics.

### CITE/seq-Count-1.4.2-foss-2016b-Python-3.7.4
A python package that allows to count antibody TAGS from a CITE-seq and/or
 cell hashing experiment.


### h5py/2.9.0-foss-2016b-Python-3.7.4
HDF5 for Python (h5py) is a general-purpose Python interface to the Hierarchical Data Format library,
 version 5. HDF5 is a versatile, mature scientific software library designed for the fast, flexible storage of enormous
 amounts of data.

### jupyter/1.0.0-foss-2016b-Python-3.7.4
JupyterLab is a web-based interactive development environment for Jupyter
 notebooks, code, and data.


### Keras/2.3.0-foss-2016b-Python-3.7.4
Keras is a minimalist, highly modular neural networks library, written in Python and
capable of running on top of either TensorFlow or Theano.


### matplotlib/3.1.1-foss-2016b-Python-3.7.4
matplotlib is a python 2D plotting library which produces publication quality figures in a variety of
 hardcopy formats and interactive environments across platforms. matplotlib can be used in python scripts, the python
 and ipython shell, web application servers, and six graphical user interface toolkits.


### megalodon/1.0.0-alpha.1-foss-2016b-Python-3.7.4
Megalodon provides "basecalling augmentation" for raw nanopore
 sequencing reads, including direct, reference-guided SNP and modified base calling.


### pkgconfig/1.3.1-foss-2016b-Python-3.7.4
pkgconfig is a Python module to interface with the pkg-config command line tool


### pybedtools/0.8.0-foss-2016b-Python-3.7.4
pybedtools wraps and extends BEDTools and offers feature-level manipulations from
 within Python.


### pybind11/2.3.0-foss-2016b-Python-3.7.4
pybind11 is a lightweight header-only library that exposes C++ types in Python and vice versa,
 mainly to create Python bindings of existing C++ code.


### Pysam/0.15.3-foss-2016b-Python-3.7.4
Pysam is a python module for reading and manipulating Samfiles. 
 It's a lightweight wrapper of the samtools C-API. Pysam also includes an interface for tabix.


### Pysamstats/1.1.2-foss-2016b-Python-3.7.4
A Python utility for calculating statistics against genome positions based on
 sequence alignments from a SAM or BAM file.


### python/fire-0.2.1-foss-2016b-Python-3.7.4
Python Fire is a library for automatically generating command line interfaces
 (CLIs) from absolutely any Python object.


### PyYAML/5.1.2-foss-2016b-Python-3.7.4
PyYAML is a YAML parser and emitter for the Python programming language.


### scikit/learn-0.21.2-foss-2016b-Python-3.7.4
Scikit-learn integrates machine learning algorithms in the tightly-knit scientific Python world,
building upon numpy, scipy, and matplotlib. As a machine-learning module,
it provides versatile tools for data mining and analysis in any field of science and engineering.
It strives to be simple and efficient, accessible to everybody, and reusable in various contexts.

### SWIG/4.0.0-foss-2016b-Python-3.7.4
SWIG is a software development tool that connects programs written in C and C++ with
 a variety of high-level programming languages.


### TensorFlow/1.14.0-foss-2016b-Python-3.7.4
An open-source software library for Machine Intelligence


### Theano/1.0.4-foss-2016b-Python-3.7.4
Theano is a Python library that allows you to define, optimize,
and evaluate mathematical expressions involving multi-dimensional arrays efficiently.


### TraCeR/0.6.0-foss-2016b-Python-3.7.4
TraCeR reconstructs the sequences of rearranged and expressed T cell receptor
 genes from single-cell RNA-seq data. It then uses the TCR sequences to identify cells that
 have the same receptor sequences and so derive from the same original clonally-expanded cell.



## Available Python Modules