<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.15.2 by Michael Rose
  Copyright 2013-2019 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Multiple Testing - Fred Hutch Biomedical Data Science Wiki</title>
<meta name="description" content="Statistically, multiple testing is defined as the testing of more than one hypothesis at a time.  Pragmatically, this changes ‘omics significance levels based on the breadth of your assay and the number of different cohort arrangements on which statistical tests are run. The resources below will help you understand the broad strokes before consulting with a statistician about your project’s specific details.">



<meta property="og:type" content="article">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Fred Hutch Biomedical Data Science Wiki">
<meta property="og:title" content="Multiple Testing">
<meta property="og:url" content="http://localhost:4000/generation/multiple_testing/">


  <meta property="og:description" content="Statistically, multiple testing is defined as the testing of more than one hypothesis at a time.  Pragmatically, this changes ‘omics significance levels based on the breadth of your assay and the number of different cohort arrangements on which statistical tests are run. The resources below will help you understand the broad strokes before consulting with a statistician about your project’s specific details.">







  <meta property="article:published_time" content="2019-04-02T18:09:25-07:00">



  <meta property="article:modified_time" content="2019-03-19T00:00:00-07:00">



  

  


<link rel="canonical" href="http://localhost:4000/generation/multiple_testing/">







  <script type="application/ld+json">
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000",
      "sameAs": null
    }
  </script>







<!-- end _includes/seo.html -->


<link href="index.html" type="application/atom+xml" rel="alternate" title="Fred Hutch Biomedical Data Science Wiki Feed">

<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">

<!--[if IE ]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->



    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
              <a href="http://localhost:4000"><img src="/images/Biomedical-Data-Science-Wiki-logo.png" alt="FH biomedical data science wiki icon" width="600" align="left|middle" hspace="0"></a>
        <!--<a class="site-title" href="/">Fred Hutch Biomedical Data Science Wiki</a>-->

        <ul class="visible-links">
          
          
          <li class="masthead__menu-item">
            <a href="/generation/gen_index/" >Data Generation</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/bioinformatics/inf_index/" >Bioinformatics</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/computing/comp_index/" >Computing</a>
          </li>
          
        </ul>
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z"
              transform="translate(-.01)"></path>
          </svg>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">
  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Data Generation</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/gen_index/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generationdemos/" class="">Resource Library</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Study Design and Funding</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/study_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/study_proposal/" class="">Proposals and Funding</a></li>
          
            
            

            
            

            <li><a href="/generation/study_hypothesis/" class="">Hypothesis Development and Big Molecular Data</a></li>
          
            
            

            
            

            <li><a href="/generation/multiple_testing/" class="active">Multiple Testing</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Consent, Privacy and Security</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/human_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/human_consentIRB/" class="">Consenting and Big Molecular Data</a></li>
          
            
            

            
            

            <li><a href="/generation/human_privacySecurity/" class="">Data Privacy and Security</a></li>
          
            
            

            
            

            <li><a href="/generation/security_legal/" class="">Compliance and Legal</a></li>
          
            
            

            
            

            <li><a href="/generation/human_deidentification/" class="">De-identification</a></li>
          
            
            

            
            

            <li><a href="/generation/human_shareDeposits/" class="">Data Sharing and Public Repositories</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Clinical and Experimental Data</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/clsp_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_clinicalCov/" class="">Clinical Data</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_specimenBanking/" class="">Specimen Banking</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_labCov/" class="">Experimental Data</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Large Scale Data Generation</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/datagen_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_platformsData/" class="">Genomics Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_flowcytometry/" class="">Flow Cytometry Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_imaging/" class="">Imaging Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_proteomics/" class="">Proteomics Data</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Other Laboratory Resources</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/labman_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_assayPrep/" class="">Assay Material Prep and QC</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_dnaApproaches/" class="">DNA-Based Approaches</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_rnaApproaches/" class="">RNA-Based Approaches</a></li>
          
            
            

            
            

            <li><a href="/generation/labman_software/" class="">Support Software</a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>
    
  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="Multiple Testing">
    <meta itemprop="description" content="Statistically, multiple testing is defined as the testing of more than one hypothesis at a time.  Pragmatically, this changes ‘omics significance levels based on the breadth of your assay and the number of different cohort arrangements on which statistical tests are run. The resources below will help you understand the broad strokes before consulting with a statistician about your project’s specific details.">
    <meta itemprop="datePublished" content="April 02, 2019">
    <meta itemprop="dateModified" content="March 19, 2019">

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Multiple Testing
</h1>
          
          <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_generation/multiple_testing.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fgeneration%2Fmultiple_testing%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->

        </header>
      


      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right ">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On This Page</h4></header>
              <ul class="toc__menu">
  <li><a href="#reading-material">Reading material</a></li>
  <li><a href="#hands-on-example-r">Hands on Example (R)</a>
    <ul>
      <li><a href="#hypothetical-study-design">Hypothetical Study Design</a></li>
      <li><a href="#some-definitions">Some definitions</a></li>
      <li><a href="#analyze-the-study-data">Analyze the Study Data</a></li>
      <li><a href="#correcting-for-multiple-testing">Correcting for Multiple testing</a></li>
    </ul>
  </li>
  <li><a href="#family-wise-error-rate-fwer">Family Wise Error Rate: FWER</a>
    <ul>
      <li><a href="#bonferroni">Bonferroni</a></li>
      <li><a href="#holm-procedure">Holm procedure</a></li>
      <li><a href="#implementation">Implementation</a></li>
    </ul>
  </li>
  <li><a href="#false-discovery-rate-fdr">False Discovery Rate: FDR</a>
    <ul>
      <li><a href="#benjamini-and-hochberg">Benjamini and Hochberg</a></li>
      <li><a href="#q-value">q-value</a></li>
      <li><a href="#implementation-of-these-fdr-methods">Implementation of these FDR methods</a></li>
    </ul>
  </li>
  <li><a href="#another-example-of-the-fwer">Another Example of the FWER</a></li>
</ul>
            </nav>
          </aside>
        
        <p>Statistically, multiple testing is defined as the testing of more than one hypothesis at a time.  Pragmatically, this changes ‘omics significance levels based on the breadth of your assay and the number of different cohort arrangements on which statistical tests are run. The resources below will help you understand the broad strokes before consulting with a statistician about your project’s specific details.</p>

<h2 id="reading-material">Reading material</h2>
<p>We aim to curate some useful reading material about multiple testing relevant to Fred Hutch studies.  Please email additional suggestions to Fred Hutch username <code class="highlighter-rouge">sciwiki</code>.</p>
<ul>
  <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2907892/">How does multiple testing correction work?</a> (Noble 2010)</li>
</ul>

<h2 id="hands-on-example-r">Hands on Example (R)</h2>
<p>This example provides information on the Family Wise Error Rate (FWER) and the False Discovery Rate (FDR), using code run in <a href="/bioinformatics/prog_R/">R</a>. We provide information on a few commonly used approaches for correcting for multiple testing as well. This is not an exhaustive list, as there is still ongoing research on how to adjust for multiple testing.</p>

<h3 id="hypothetical-study-design">Hypothetical Study Design</h3>
<p>For the sake of having an example on hand, let’s draw up a hypothetical study. Imagine we have observed gene expression for 1000 independent transcripts in 500 study participants. We are interested in testing if there is any association between any of those 1000 transcripts and reading this wiki page. For each of the 500 people, we have an indicator (1 or 0) for whether they have read the wiki page you are reading right now. For each gene we are testing the following hypothesis:</p>

<blockquote>
  <p>H<sub>0j</sub>: The gene expression at transcript j is not associated with having read this wiki.</p>

  <p>H<sub>1j</sub>: The gene expression at transcript j is associated with having read this wiki.</p>
</blockquote>

<p>H<sub>0j</sub> is the NULL hypothesis for transcript j and H<sub>1j</sub> is the alternative. For each test we have a p-value which we use as evidence against H<sub>0j</sub>.</p>

<h3 id="some-definitions">Some definitions</h3>
<p>Before continuing, it is useful to define some statistical terms.</p>
<blockquote>
  <p>Type I error: Probability that we incorrectly  reject H<sub>0j</sub></p>
</blockquote>

<p>This mean we observe a p-value that would lead us to reject the NULL hypothesis and give us evidence to <strong>suggest</strong> the alternative. Often, we want to control the Type I error at some pre-specified level α. For example, 0.05 is often used. With this threshold, if the p-value is less than 0.05, we reject the NULL hypothesis. The value of 0.05 is arbitrary and a significance cutoff should be decided prior to running any analyses and should be based on prevalent thought within the field.</p>

<blockquote>
  <p>FWER: Family Wise Error Rate: The Probability of making at least one Type I error</p>

  <p>FDR: False Discovery Rate: Rate of tests that come back as significant that are actually false.</p>
</blockquote>

<h3 id="analyze-the-study-data">Analyze the Study Data</h3>

<p>Now let us run this hypothetical study and see what happens. We provide code below to run, if you would like to follow along in R. We generate the gene expression data (Y) and the exposure information (X) of whether they have read this wiki page. We’re going to use as our significance threshold 0.05. Thus if the p-value is less than 0.05, we are going to reject the NULL hypothesis that there is no association between gene expression at that gene and reading this wiki page.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">set.seed</span><span class="p">(</span><span class="m">20190101</span><span class="p">)</span><span class="w">

</span><span class="n">N</span><span class="o">&lt;</span><span class="m">-500</span><span class="w"> </span><span class="c1">#the sample size</span><span class="w">
</span><span class="n">M</span><span class="o">&lt;</span><span class="m">-1000</span><span class="w"> </span><span class="c1">#number of "genes" we are testing</span><span class="w">
</span><span class="n">X</span><span class="o">&lt;-</span><span class="n">rbinom</span><span class="p">(</span><span class="n">N</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">0.2</span><span class="p">)</span><span class="w"> </span><span class="c1"># Variable of interest</span><span class="w">
</span><span class="n">Y</span><span class="o">&lt;-</span><span class="n">replicate</span><span class="p">(</span><span class="n">M</span><span class="p">,</span><span class="n">rnorm</span><span class="p">(</span><span class="n">N</span><span class="p">))</span><span class="w"> </span><span class="c1"># The observed gene expression</span><span class="w">
</span><span class="n">TestIt</span><span class="o">&lt;-</span><span class="n">matrix</span><span class="p">(</span><span class="n">nrow</span><span class="o">=</span><span class="n">M</span><span class="p">,</span><span class="n">ncol</span><span class="o">=</span><span class="m">4</span><span class="p">)</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">){</span><span class="w">
  </span><span class="n">R</span><span class="m">1</span><span class="o">&lt;-</span><span class="n">summary</span><span class="p">(</span><span class="n">lm</span><span class="p">(</span><span class="n">Y</span><span class="p">[,</span><span class="n">i</span><span class="p">]</span><span class="o">~</span><span class="n">X</span><span class="p">))</span><span class="w">
  </span><span class="n">TestIt</span><span class="p">[</span><span class="n">i</span><span class="p">,]</span><span class="o">&lt;-</span><span class="n">coef</span><span class="p">(</span><span class="n">R</span><span class="m">1</span><span class="p">)[</span><span class="m">2</span><span class="p">,]</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<p>If we look at the p-values (which are the fourth column of the R matrix TestIt above), we can count how many NULL hypothesis we would reject.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">sum</span><span class="p">(</span><span class="n">TestIt</span><span class="p">[,</span><span class="m">4</span><span class="p">]</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 58
</code></pre></div></div>

<p>We can see that we improperly reject 58 tests solely by chance. This is not ideal. The reason for this is that when we are under the NULL and we are using a well-behaving test, the p-values follow something called a continuous uniform distribution from 0 to 1. Under this distribution, the probability that one p-value is less than 0.05 is 0.05, which is fine if we are only performing one test. However, when we are performing 1000 tests, the probability that <em>at least one of those p-values is less than 0.05</em> is approximately 1. IE our FWER under this case is going to be 1.</p>

<h3 id="correcting-for-multiple-testing">Correcting for Multiple testing</h3>
<p>We now get into the details of how to correct for multiple testing. There are two ways one can go about doing this, control the FWER or the FDR. Which one is used, should be decided prior to the analysis.</p>

<p>Consider the following table for our example (borrowed from Wikipedia).</p>

<table>
  <thead>
    <tr>
      <th> </th>
      <th style="text-align: center">H<sub>0j</sub> is true</th>
      <th style="text-align: right">H<sub>1j</sub> is True</th>
      <th style="text-align: right">Total</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Test is declared significant</td>
      <td style="text-align: center">V</td>
      <td style="text-align: right">S</td>
      <td style="text-align: right">V+S</td>
    </tr>
    <tr>
      <td>Test is not declared significant</td>
      <td style="text-align: center">U</td>
      <td style="text-align: right">T</td>
      <td style="text-align: right">U+T</td>
    </tr>
    <tr>
      <td>Totals</td>
      <td style="text-align: center">V+U</td>
      <td style="text-align: right">S+T</td>
      <td style="text-align: right">1000</td>
    </tr>
  </tbody>
</table>

<h2 id="family-wise-error-rate-fwer">Family Wise Error Rate: FWER</h2>

<p>Remember, the FWER is the probability that we perform at least one type I error in our analysis. This is the probability that the cell V is greater than or equal to 1 (i.e., the test is declared significant  but  H<sub>0j</sub> is true). When a study says they are controlling  the FWER at some level α, what that means is that:</p>

<blockquote>
  <p>P(V≥1)≤α</p>
</blockquote>

<p>Let us say that we originally decided to say the test is significant if the p-value is less than 0.05. In that situation:</p>

<blockquote>
  <p>P(V≥1)≥0.05</p>
</blockquote>

<p>Therefore, our FWER is not controlled at the rate of 0.05. We actually have more than a 5% chance of making a Type I error, which is not we want. If all 1000 tests are truly null, the probability of making at least one error is equal to 1-(1-0.05)^1000 or approximately 1.</p>

<p>Below we give two different approaches to control for the FWER. This is by no means exhaustive and there are other approaches out there. All two are implemented in the R function p.adjust.</p>

<h3 id="bonferroni">Bonferroni</h3>

<p>The most widely used approach. A test is declared significance if the corresponding p-value is less than α divided by the number of tests. In our example, we would use 0.05/1000 as opposed to 0.05. Here would have:</p>

<blockquote>
  <p>P(V≥1)=P(Any of the H<sub>0j</sub> are improperly rejected)≤(V+U)0.05/1000≤(1000)0.05/(1000)=0.05</p>
</blockquote>

<p>Therefore we have that the P(V≥1)≤ 0.05, the desired result.</p>

<h3 id="holm-procedure">Holm procedure</h3>

<p>The Holm correction is a stepwise procedure, where the significance threshold changes as you reject tests.</p>

<ol>
  <li>Look smallest p-value. If less than α/#tests reject the respective H<sub>0j</sub>.</li>
  <li>Look at next smallest p-value. If this p-value is less than α/(#tests-1) reject H<sub>0j</sub>.</li>
  <li>Continue looking at successively larger p-values in the same manner until you can no longer reject.</li>
</ol>

<h3 id="implementation">Implementation</h3>
<p>Let us implement these in our example and see how they do. First the Bonferroni correction. The function p.adjust gives back a corrected p-value.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">BonPval</span><span class="o">&lt;-</span><span class="n">p.adjust</span><span class="p">(</span><span class="n">TestIt</span><span class="p">[,</span><span class="m">4</span><span class="p">],</span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"bonferroni"</span><span class="p">)</span><span class="w">
</span><span class="nf">sum</span><span class="p">(</span><span class="n">BonPval</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 0
</code></pre></div></div>

<p>So we now no longer reject any tests. Next we do the Holm procedure:</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">HolmPval</span><span class="o">&lt;-</span><span class="n">p.adjust</span><span class="p">(</span><span class="n">TestIt</span><span class="p">[,</span><span class="m">4</span><span class="p">],</span><span class="n">method</span><span class="o">=</span><span class="s2">"holm"</span><span class="p">)</span><span class="w">
</span><span class="nf">sum</span><span class="p">(</span><span class="n">HolmPval</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 0 And we get the same result.
</code></pre></div></div>

<h2 id="false-discovery-rate-fdr">False Discovery Rate: FDR</h2>

<p>The False Discovery Rate is a newer correction approach developed by Benjamini and Hochberg in the 1990’s. The goal of the FDR is to identify as many significant tests while controlling the number of false positives identified. This means we want to keep V/(V+S) from the above table below a certain proportion. The FDR in general is a less conservative approach to multiple testing than the FWER (though they are controlling separate things).  In general, if something is significant via the FWER at a level α, it will be significant at the FDR level α as well.</p>

<p>Below we give a few approaches, but again these are by no means conclusive. This is still an ongoing area of research in general.</p>

<h3 id="benjamini-and-hochberg">Benjamini and Hochberg</h3>

<p>If we are controlling at the level α, we order the p-values and find the k<sup>th</sup> ordered p-value such that it is less than α times k/m. This method is implemented within the p.adjust function in R.</p>

<h3 id="q-value">q-value</h3>

<p>A popular approach in bioinformatics, the q-value is a Bayesian approach to the FDR developed by John Storey. It is implemented in the Bioconductor R package q-value.</p>

<h3 id="implementation-of-these-fdr-methods">Implementation of these FDR methods</h3>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">BHPval</span><span class="o">&lt;-</span><span class="n">p.adjust</span><span class="p">(</span><span class="n">TestIt</span><span class="p">[,</span><span class="m">4</span><span class="p">],</span><span class="n">method</span><span class="o">=</span><span class="s2">"BH"</span><span class="p">)</span><span class="w">
</span><span class="nf">sum</span><span class="p">(</span><span class="n">BHPval</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 0
</code></pre></div></div>

<p>And now qvalue</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">qvalue</span><span class="p">)</span><span class="w">
</span><span class="n">TheQval</span><span class="o">&lt;-</span><span class="n">qvalue</span><span class="p">(</span><span class="n">TestIt</span><span class="p">[,</span><span class="m">4</span><span class="p">])</span><span class="o">$</span><span class="n">qv</span><span class="w">
</span><span class="nf">sum</span><span class="p">(</span><span class="n">TheQval</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 0
</code></pre></div></div>

<p>Similar to when we corrected for the FWER we no longer reject any tests.</p>

<h2 id="another-example-of-the-fwer">Another Example of the FWER</h2>

<p>Now let us run some examples to see what happens when we run this study over and over again to get an estimate of what the FWER is under these scenarios. This will tell us how well these testing procedures are really doing.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">Numstudies</span><span class="o">&lt;</span><span class="m">-5000</span><span class="w"> </span><span class="c1"># Number of studies</span><span class="w">
</span></code></pre></div></div>

<p>Define a function to use that will do our analyses a little quicker so we do not have to call lm.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">fastReg.func</span><span class="o">&lt;-</span><span class="k">function</span><span class="p">(</span><span class="n">A</span><span class="p">,</span><span class="n">nY</span><span class="p">){</span><span class="w">
  </span><span class="n">XX</span><span class="o">&lt;-</span><span class="n">cbind</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="n">A</span><span class="p">)</span><span class="w">
  </span><span class="n">EY</span><span class="o">&lt;-</span><span class="n">solve</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">XX</span><span class="p">)</span><span class="o">%*%</span><span class="n">XX</span><span class="p">)</span><span class="o">%*%</span><span class="n">t</span><span class="p">(</span><span class="n">XX</span><span class="p">)</span><span class="o">%*%</span><span class="n">nY</span><span class="w">
  </span><span class="n">residY</span><span class="o">&lt;-</span><span class="n">nY</span><span class="o">-</span><span class="n">XX</span><span class="o">%*%</span><span class="n">EY</span><span class="w">
  </span><span class="n">SigEst</span><span class="o">&lt;-</span><span class="n">colSums</span><span class="p">(</span><span class="n">residY</span><span class="o">^</span><span class="m">2</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">A</span><span class="p">)</span><span class="o">-</span><span class="n">ncol</span><span class="p">(</span><span class="n">XX</span><span class="p">))</span><span class="w">
  </span><span class="n">VarEst</span><span class="o">&lt;-</span><span class="n">SigEst</span><span class="o">*</span><span class="n">solve</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">XX</span><span class="p">)</span><span class="o">%*%</span><span class="n">XX</span><span class="p">)[</span><span class="m">2</span><span class="p">,</span><span class="m">2</span><span class="p">]</span><span class="w">
  </span><span class="n">Pval.A.to.Y</span><span class="o">&lt;</span><span class="m">-2</span><span class="o">*</span><span class="n">pt</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">EY</span><span class="p">[</span><span class="m">2</span><span class="p">,]</span><span class="o">/</span><span class="nf">sqrt</span><span class="p">(</span><span class="n">VarEst</span><span class="p">)),</span><span class="n">df</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">A</span><span class="p">)</span><span class="o">-</span><span class="n">ncol</span><span class="p">(</span><span class="n">XX</span><span class="p">),</span><span class="n">lower.tail</span><span class="o">=</span><span class="nb">F</span><span class="p">)</span><span class="w">
  </span><span class="nf">return</span><span class="p">(</span><span class="n">cbind</span><span class="p">(</span><span class="n">EY</span><span class="p">[</span><span class="m">2</span><span class="p">,],</span><span class="n">Pval.A.to.Y</span><span class="p">))</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<p>And run the simulation</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">set.seed</span><span class="p">(</span><span class="m">44511</span><span class="p">)</span><span class="w">
</span><span class="n">EachStudy</span><span class="o">&lt;-</span><span class="n">matrix</span><span class="p">(</span><span class="n">nrow</span><span class="o">=</span><span class="n">Numstudies</span><span class="p">,</span><span class="n">ncol</span><span class="o">=</span><span class="m">4</span><span class="p">)</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">k</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">Numstudies</span><span class="p">){</span><span class="w">
  </span><span class="n">X</span><span class="o">&lt;-</span><span class="n">rbinom</span><span class="p">(</span><span class="n">N</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">0.2</span><span class="p">)</span><span class="w"> </span><span class="c1"># Variable of interest</span><span class="w">
  </span><span class="n">Y</span><span class="o">&lt;-</span><span class="n">replicate</span><span class="p">(</span><span class="n">M</span><span class="p">,</span><span class="n">rnorm</span><span class="p">(</span><span class="n">N</span><span class="p">))</span><span class="w">
  </span><span class="n">AA</span><span class="o">&lt;-</span><span class="n">fastReg.func</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">Y</span><span class="p">)</span><span class="w">
  </span><span class="n">EachStudy</span><span class="p">[</span><span class="n">k</span><span class="p">,</span><span class="m">1</span><span class="p">]</span><span class="o">&lt;-</span><span class="nf">sum</span><span class="p">(</span><span class="n">AA</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
  </span><span class="n">B</span><span class="m">1</span><span class="o">&lt;-</span><span class="n">p.adjust</span><span class="p">(</span><span class="n">AA</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span><span class="s2">"bonferroni"</span><span class="p">)</span><span class="w">
  </span><span class="n">EachStudy</span><span class="p">[</span><span class="n">k</span><span class="p">,</span><span class="m">2</span><span class="p">]</span><span class="o">&lt;-</span><span class="nf">sum</span><span class="p">(</span><span class="n">B</span><span class="m">1</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
  </span><span class="n">H</span><span class="m">1</span><span class="o">&lt;-</span><span class="n">p.adjust</span><span class="p">(</span><span class="n">AA</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span><span class="s2">"holm"</span><span class="p">)</span><span class="w">
  </span><span class="n">EachStudy</span><span class="p">[</span><span class="n">k</span><span class="p">,</span><span class="m">3</span><span class="p">]</span><span class="o">&lt;-</span><span class="nf">sum</span><span class="p">(</span><span class="n">H</span><span class="m">1</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
  </span><span class="n">BH1</span><span class="o">&lt;-</span><span class="n">p.adjust</span><span class="p">(</span><span class="n">AA</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span><span class="s2">"BH"</span><span class="p">)</span><span class="w">
  </span><span class="n">EachStudy</span><span class="p">[</span><span class="n">k</span><span class="p">,</span><span class="m">4</span><span class="p">]</span><span class="o">&lt;-</span><span class="nf">sum</span><span class="p">(</span><span class="n">BH1</span><span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>
<p>In the above, we are saving from each study of 1000 genes four things.
1) How many uncorrected p-values are less than 0.05 (first column of the matrix EachStudy).
2) How many Bonferroni corrected p-values are less than 0.05 (second column of EachStudy). This is equivalent to how many uncorrected p-values are less than 0.05/1000.
3) How many Holm adjusted p-values are less than 0.05 (third column of EachStudy).
4) How many Benjamini Hochberg corrected p-values are less than 0.05 (fourth column of EachStudy). This is controlling the FDR.</p>

<p>We can now look at what the FWER is if we didn’t perform any correction at all by taking the mean of how many of these 5000 studies resulted in at least one p-value less than 0.05.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">mean</span><span class="p">(</span><span class="n">EachStudy</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 1
</code></pre></div></div>

<p>It is 1 as we expected. Next we can look at how the Bonferroni and Holm corrected approach do by mean of how many of these 5000 studies resulted in at least one p-value less than 0.05.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">mean</span><span class="p">(</span><span class="n">EachStudy</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 0.0488
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">mean</span><span class="p">(</span><span class="n">EachStudy</span><span class="p">[,</span><span class="m">3</span><span class="p">]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 0.0488
</code></pre></div></div>

<p>We see that it is less than 0.05 (but still close) which is the desired result. Finally, we look at the FDR and we see a similar result (though now we are controlling the FDR).</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">mean</span><span class="p">(</span><span class="n">EachStudy</span><span class="p">[,</span><span class="m">4</span><span class="p">]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] 0.05
</code></pre></div></div>

        
      </section>

      <footer class="page__meta">
        
        


        
          <p class="page__date"><strong><i class="fas fa-fw fa-calendar-alt" aria-hidden="true"></i> Updated:</strong> <time datetime="2019-03-19">March 19, 2019</time></p>
        
      </footer>

      <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_generation/multiple_testing.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fgeneration%2Fmultiple_testing%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->


      
  <nav class="pagination">
    
      <a href="/generation/labman_software/" class="pagination--pager" title="Software for Research and Laboratory Management
">Previous</a>
    
    
      <a href="/generation/security_legal/" class="pagination--pager" title="Compliance and Legal Agreements
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form onsubmit="return googleCustomSearchExecute();" id="cse-search-box-form-id">
    <input type="text" id="cse-search-input-box-id" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
    </form>
    <div id="results" class="results">
        <gcse:searchresults-only></gcse:searchresults-only>    
    </div></div>

      </div>
    

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
          <li><a href="/contributors/"><i class="fas fa-user-edit"></i> See our Contributors</a></li>

          <li><a href="https://github.com/FredHutch/wiki"><i class="fab fa-fw fa-github"></i> See our GitHub Repo</a></li>

          <li><a href="https://fhbig.slack.com/"><i class="fab fa-fw fa-slack"> </i> Join the Conversation on Slack</a></li>

          <li><a href="https://fredhutch.github.io/FHBig/"><img src="/images/FHBig.png" alt="fhbig" width="40" align="left|middle" hspace="0"> Find Upcoming Events</a></li>

          <li><a href="https://www.fredhutch.io/"><img src="/images/fhio.png" alt="fredhutchio" width="20" align="left|middle" hspace="0"> Find Training</a></li>


  </ul>
</div>

<div class="page__footer-copyright">&copy; 2019 Fred Hutch Biomedical Data Science Wiki. Powered by <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>
  <script defer src="https://use.fontawesome.com/releases/v5.7.1/js/all.js" integrity="sha384-eVEQC9zshBn0rFj4+TU78eNA19HMNigMviK/PU/FFjLXqa/GKPgX58rvt5Z8PLs7" crossorigin="anonymous"></script>


<script>
  (function () {
    var cx = '018020560330782233421:waz3ch9w_wg';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();

  function googleCustomSearchExecute() {
    var input = document.getElementById('cse-search-input-box-id');
    var element = google.search.cse.element.getElement('searchresults-only0');
    if (input.value == '') {
      element.clearAllResults();
    } else {
      element.execute(input.value);
    }
    return false;
  }

  
</script>





  </body>
</html>
