<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.15.2 by Michael Rose
  Copyright 2013-2019 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Data Sharing and Public Repositories - Fred Hutch Biomedical Data Science Wiki</title>
<meta name="description" content="Whether required for funding and publication or desired for its beneficial impact on research progress, understanding the why and how of data sharing is essential for modern biomedical research programs.">



<meta property="og:type" content="article">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Fred Hutch Biomedical Data Science Wiki">
<meta property="og:title" content="Data Sharing and Public Repositories">
<meta property="og:url" content="http://localhost:4000/generation/human_shareDeposits/">


  <meta property="og:description" content="Whether required for funding and publication or desired for its beneficial impact on research progress, understanding the why and how of data sharing is essential for modern biomedical research programs.">







  <meta property="article:published_time" content="2019-04-02T18:09:25-07:00">



  <meta property="article:modified_time" content="2019-04-01T00:00:00-07:00">



  

  


<link rel="canonical" href="http://localhost:4000/generation/human_shareDeposits/">







  <script type="application/ld+json">
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000",
      "sameAs": null
    }
  </script>







<!-- end _includes/seo.html -->


<link href="index.html" type="application/atom+xml" rel="alternate" title="Fred Hutch Biomedical Data Science Wiki Feed">

<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">

<!--[if IE ]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->



    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
              <a href="http://localhost:4000"><img src="/images/Biomedical-Data-Science-Wiki-logo.png" alt="FH biomedical data science wiki icon" width="600" align="left|middle" hspace="0"></a>
        <!--<a class="site-title" href="/">Fred Hutch Biomedical Data Science Wiki</a>-->

        <ul class="visible-links">
          
          
          <li class="masthead__menu-item">
            <a href="/generation/gen_index/" >Data Generation</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/bioinformatics/inf_index/" >Bioinformatics</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/computing/comp_index/" >Computing</a>
          </li>
          
        </ul>
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z"
              transform="translate(-.01)"></path>
          </svg>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">
  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Data Generation</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/gen_index/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generationdemos/" class="">Resource Library</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Study Design and Funding</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/study_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/study_proposal/" class="">Proposals and Funding</a></li>
          
            
            

            
            

            <li><a href="/generation/study_hypothesis/" class="">Hypothesis Development and Big Molecular Data</a></li>
          
            
            

            
            

            <li><a href="/generation/multiple_testing/" class="">Multiple Testing</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Consent, Privacy and Security</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/human_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/human_consentIRB/" class="">Consenting and Big Molecular Data</a></li>
          
            
            

            
            

            <li><a href="/generation/human_privacySecurity/" class="">Data Privacy and Security</a></li>
          
            
            

            
            

            <li><a href="/generation/security_legal/" class="">Compliance and Legal</a></li>
          
            
            

            
            

            <li><a href="/generation/human_deidentification/" class="">De-identification</a></li>
          
            
            

            
            

            <li><a href="/generation/human_shareDeposits/" class="active">Data Sharing and Public Repositories</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Clinical and Experimental Data</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/clsp_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_clinicalCov/" class="">Clinical Data</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_specimenBanking/" class="">Specimen Banking</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_labCov/" class="">Experimental Data</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Large Scale Data Generation</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/datagen_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_platformsData/" class="">Genomics Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_flowcytometry/" class="">Flow Cytometry Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_imaging/" class="">Imaging Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_proteomics/" class="">Proteomics Data</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Other Laboratory Resources</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/labman_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_assayPrep/" class="">Assay Material Prep and QC</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_dnaApproaches/" class="">DNA-Based Approaches</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_rnaApproaches/" class="">RNA-Based Approaches</a></li>
          
            
            

            
            

            <li><a href="/generation/labman_software/" class="">Support Software</a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>
    
  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="Data Sharing and Public Repositories">
    <meta itemprop="description" content="Whether required for funding and publication or desired for its beneficial impact on research progress, understanding the why and how of data sharing is essential for modern biomedical research programs.">
    <meta itemprop="datePublished" content="April 02, 2019">
    <meta itemprop="dateModified" content="April 01, 2019">

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Data Sharing and Public Repositories
</h1>
          
          <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_generation/human_shareDeposits.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fgeneration%2Fhuman_shareDeposits%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->

        </header>
      


      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right ">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On This Page</h4></header>
              <ul class="toc__menu">
  <li><a href="#benefits-of-data-sharing">Benefits of Data Sharing</a></li>
  <li><a href="#data-sharing-and-management-plans">Data Sharing and Management Plans</a>
    <ul>
      <li><a href="#reading-material">Reading Material</a></li>
    </ul>
  </li>
  <li><a href="#data-sharing-responsibilities">Data Sharing Responsibilities</a></li>
  <li><a href="#required-data-sharing">Required Data Sharing</a>
    <ul>
      <li><a href="#clinicaltrialsgov">ClinicalTrials.gov</a></li>
      <li><a href="#funding-agencies">Funding Agencies</a></li>
      <li><a href="#data-sharing-as-part-of-publications">Data Sharing as Part of Publications</a></li>
    </ul>
  </li>
  <li><a href="#genomic-data-sharing">Genomic Data Sharing</a>
    <ul>
      <li><a href="#nih-genomic-data-sharing-gds-policy">NIH Genomic Data Sharing (GDS) Policy</a></li>
      <li><a href="#nih-public-genomic-data-repositories">NIH Public Genomic Data Repositories</a></li>
    </ul>
  </li>
  <li><a href="#some-public-repositories">Some Public Repositories</a></li>
</ul>
            </nav>
          </aside>
        
        <p>Whether required for funding and publication or desired for its beneficial impact on research progress, understanding the why and how of data sharing is essential for modern biomedical research programs.</p>

<h2 id="benefits-of-data-sharing">Benefits of Data Sharing</h2>

<p>Some of the benefits of data sharing include:</p>

<ul>
  <li>
    <p>reinforcing open scientific inquiry</p>
  </li>
  <li>
    <p>encouraging diversity of analysis and opinion</p>
  </li>
  <li>
    <p>promoting new research, testing of new or alternative hypotheses and methods of analysis</p>
  </li>
  <li>
    <p>supporting studies on data collection methods and measurement</p>
  </li>
  <li>
    <p>facilitating education of new researchers</p>
  </li>
  <li>
    <p>enabling the exploration of topics not envisioned by the initial investigators</p>
  </li>
  <li>
    <p>permitting the creation of new data sets by combining data from multiple sources</p>
  </li>
  <li>
    <p>gaining for individual investigators additional insights from other investigators’ studies of the data</p>
  </li>
  <li>
    <p>increasing the visibility and credibility of one's own research based on the data</p>
  </li>
  <li>
    <p>opening opportunities for developing new collaborations and for access to complementary data sets</p>
  </li>
</ul>

<h2 id="data-sharing-and-management-plans">Data Sharing and Management Plans</h2>

<p>A good resource for data sharing management plan development is <a href="https://dmptool.org/">DMPTool</a> sponsored by the University of California. DMPTool lets data owners create, review and share data management plans and provides ancillary knowledge to support sharing and management.</p>

<p>The following should be considered when developing a data sharing plan:</p>

<ol>
  <li>
    <p>Data Description: What data will be generated? How will you create the data? (simulated, observed, experimental, software, physical collections)</p>
  </li>
  <li>
    <p>Existing Data: Will you be using existing data? What is the relationship between the data you are collecting and existing data?</p>
  </li>
  <li>
    <p>Audience: Who will potentially use the data?</p>
  </li>
  <li>
    <p>Confidentiality: What coding, systems, and gatekeeping procedures are in place to protect the confidentiality of the data subjects?</p>
  </li>
  <li>
    <p>Access and Sharing: How will data files be shared? How will others access them?</p>
  </li>
  <li>
    <p>Formats: What data formats will you be creating?</p>
  </li>
  <li>
    <p>Metadata and Documentation: What documentation will you provide to describe the data? Metadata formats and standards?</p>
  </li>
  <li>
    <p>Storage, backup, replication, versioning: Are the data files backed up regularly? Are there replicas in different locations? Are older versions of the data kept?</p>
  </li>
  <li>
    <p>Security: Are the system and storage that will be used secure?</p>
  </li>
  <li>
    <p>Budget: Any costs for preparing the data? Costs for storage and long-term access?</p>
  </li>
  <li>
    <p>Privacy, Intellectual Property: Does the data contain private or confidential information? Any copyrights?</p>
  </li>
  <li>
    <p>Archiving, Preservation, Long-term Access: What plans do you have to archive the data and other research products? Will it have long-term accessibility?</p>
  </li>
  <li>
    <p>Adherence: How will you check for adherence of this plan?</p>
  </li>
</ol>

<h3 id="reading-material">Reading Material</h3>

<ul>
  <li>
    <p><a href="http://best-practices.dataverse.org/data-management/index.html">Dataverse</a> has basic best practices guidance for data management plans.</p>
  </li>
  <li>
    <p><a href="https://dmptool.org/">DMPTool</a> from University of California</p>
  </li>
  <li>
    <p><a href="https://guides.library.duke.edu/c.php?g=633433&amp;p=4429249">Duke University Data Management Guide</a></p>
  </li>
  <li>
    <p><a href="https://library.stanford.edu/research/data-management-services/data-management-plans">Stanford University Library</a> includes some sample DMPs from NIH and other resources.</p>
  </li>
  <li>
    <p><a href="https://www.niaid.nih.gov/research/sample-data-sharing-plan">NIAID Sample Data Sharing Plan</a></p>
  </li>
  <li>
    <p><a href="https://fairsharing.org/policies/">FAIRSharing policies</a>:  A catalogue of data preservation, management and sharing policies form international funding agencies, regulators and journals.</p>
  </li>
</ul>

<h2 id="data-sharing-responsibilities">Data Sharing Responsibilities</h2>

<p>The context of a data sharing project can mandate certain safety requirements and restrictions. Some of these considerations may include whether the data is shared pre- or post- publication, or whether the data is shared locally within the Fred Hutch Cancer Consortium or to a larger national or international audience.</p>

<p>While the underlying goal of sharing is to further research and its benefits, the underlying tenets of data sharing processes are to ensure:</p>

<ul>
  <li>
    <p>Identification of who shall have access to the data</p>
  </li>
  <li>
    <p>Data sharing does not compromise individual subjects’ rights and privacy, regardless of whether the data have been used in a publication</p>
  </li>
  <li>
    <p>Data shared is restricted to only that data appropriate for a specific line of inquiry</p>
  </li>
  <li>
    <p>The integrity and quality of shared data are preserved</p>
  </li>
  <li>
    <p>Data sharing is done within the legal requirements of both sender and receiver</p>
  </li>
  <li>
    <p>Shared data is easily readable or accessible by the receiver. Consideration can be made whether metadata should be provided along with the data to make it easily understood</p>
  </li>
  <li>
    <p>Appropriate intellectual property, copyright, licensing issues are considered prior to sharing</p>
  </li>
</ul>

<p>Before sharing data, it is prudent to determine if sharing is permitted.
Below are some relevant questions to help make this determination:</p>

<ul>
  <li>
    <p>If the data is derived from human subjects research, does the associated IRB-approved informed consent (or waiver of informed consent) permit disclosure for the contemplated DUA purpose? If not, new IRB review, and a waiver of consent or re-consenting of subjects, may be required before sharing is permitted.</p>
  </li>
  <li>
    <p>If the data was collected pursuant to a sponsored research project, has the sponsor placed restrictions on the subsequent transfer of the data? Stipulations may prevent sharing or require specific sharing restrictions.</p>
  </li>
  <li>
    <p>If the data was initially received from, or derived from, data received from a third party pursuant to a contract, does that contract place restrictions on the subsequent transfer of the data? Stipulations may prevent sharing or require specific sharing restrictions.</p>
  </li>
</ul>

<h2 id="required-data-sharing">Required Data Sharing</h2>

<h3 id="clinicaltrialsgov">ClinicalTrials.gov</h3>

<p>HHS developed a final rule, Clinical Trials Registration and Results
Information Submission, made publicly available on September 16, 2016;
and the NIH issued a complementary final policy, under which NIH-funded
awardees and investigators are required to submit registration and
results information for all NIH-funded clinical trials, whether or not
the trials are covered by the FDAAA requirements.</p>

<ul>
  <li>
    <p>More information about the specific <a href="https://grants.nih.gov/policy/clinical-trials/reporting/understanding.htm">policies related to ClinicalTrials.gov</a></p>
  </li>
  <li>
    <p>Why <a href="https://clinicaltrials.gov/ct2/manage-recs/background">register and report results</a> to ClinicalTrials.gov?</p>
  </li>
</ul>

<h3 id="funding-agencies">Funding Agencies</h3>

<p>Many (but not all) federal funding agencies require data management
plans and data sharing plans as part of a grant proposal application.
Persons seeking to make a grant application to a federal funding source
should ascertain the data sharing plan requirements.</p>

<p>Some of the US federal granting agencies that require funded projects to
provide some form of data management planning include NIH, NSF, DOE,
DOD, HHS, and FDA.</p>

<p>Funding agencies may want to know:</p>

<ul>
  <li>
    <p>What data you are producing that will be accessible (shared) with
others?</p>
  </li>
  <li>
    <p>When and how will you make it discoverable? Place in a repository,
data commons or other location so others can find it?</p>
  </li>
  <li>
    <p>How will you make accessible? Will there be restrictions to some or
all the data and if so how do people obtain access?</p>
  </li>
  <li>
    <p>How will you make it useable/reusable? Will there be documentation,
definitions, descriptions of methodology, use of standard terminology,
etc.</p>
  </li>
</ul>

<h3 id="data-sharing-as-part-of-publications">Data Sharing as Part of Publications</h3>

<p>As of 1 July 2018, manuscripts submitted to journals within the
International Committee of Medical Journal Editors
(<em><a href="http://www.icmje.org/">ICJME</a></em>) reporting clinical trial
data must contain a data sharing statement, indicating:</p>

<ul>
  <li>
    <p>whether the authors intend to share individual de-identified participant data</p>
  </li>
  <li>
    <p>what specific data they intend to share</p>
  </li>
  <li>
    <p>what other study-related documents will be made available</p>
  </li>
  <li>
    <p>how the data will be accessible</p>
  </li>
  <li>
    <p>when and for how long they will be made available.</p>
  </li>
</ul>

<p>The statement may then be taken into account by ICMJE editors when considering
the paper for publication. Furthermore, clinical trials that begin enrolling
participants on or after 1 January 2019 must include a data sharing plan in the
trial’s registration if they wish to publish results in ICMJE journals. Any
deviations from this plan must be disclosed in the data sharing statement when
published.</p>

<p>ICMJE believe that scientists have a moral obligation to share clinical data to
maximize the knowledge obtained from these research efforts. The committee is
working with a number of groups to solve the various practical issues that they
acknowledge still exist to in order to achieve their goal of universal data
sharing.</p>

<h2 id="genomic-data-sharing">Genomic Data Sharing</h2>

<p>American College of Medical Genetics and Genomics Board of Directors
suggests that broad data sharing is necessary and will improve care by
making available the best data possible, by which:</p>

<ul>
  <li>
    <p>Key clinical attributes of the phenotype of those with genetic diseases can be described;</p>
  </li>
  <li>
    <p>The qualitative strength of the association between genetic diseases and the underlying causative genes can be established;</p>
  </li>
  <li>
    <p>The classification of genomic variants across the range of benign to pathogenic can be established;</p>
  </li>
  <li>
    <p>Differences in variant interpretation among laboratories can be reconciled;</p>
  </li>
  <li>
    <p>The appropriate classification of variants of uncertain significance can be made; and</p>
  </li>
  <li>
    <p>Standards used in variant classification can be improved.</p>
  </li>
</ul>

<p>As noted by the NIH and others, the nature of genomic data requires
several specific considerations be kept in mind. In addition to being
personal and unique to each individual, genomic data may, for example:</p>

<ul>
  <li>
    <p>Be stored and used indefinitely.</p>
  </li>
  <li>
    <p>Inform individuals about susceptibility to a broad range of conditions (some of which are unexpected given personal or family history).</p>
  </li>
  <li>
    <p>Carry with them risks that are uncertain or unclear.</p>
  </li>
  <li>
    <p>Be reinterpreted and change in relevance over time.</p>
  </li>
  <li>
    <p>Raise privacy concerns (in part because of the risk of re-identification).</p>
  </li>
  <li>
    <p>Be relevant for family members and reproductive decision-making.</p>
  </li>
  <li>Transform the personal decision of sharing genetic data into one with</li>
  <li>familial current and future ramifications.</li>
</ul>

<p>There are many challenges to sharing phenotypic and genotypic data. As such,
some genomic data sharing best practices are listed below:</p>

<ul>
  <li>
    <p>Investigator(s) should use requested datasets solely in connection with
the research project described in the approved data request, protocol or
other vehicle that bounds the use and disclosure of the data.</p>
  </li>
  <li>
    <p>Investigator(s) will make no attempt to identify or contact individual
participants from whom these data were collected without appropriate
approvals from the relevant IRBs.</p>
  </li>
  <li>
    <p>Investigator(s) will use informed consent documents with specific language
addressing the potential persistent harms identified data can carry, the
rights of patients to withdraw consent, the type of data involved and the
respect and trust of participants who give researchers the right to use
their data and samples for “future unspecified research.”</p>
  </li>
  <li>
    <p>Investigator(s) will not distribute these data to any entity or individual
beyond those specified in the approved data request, protocol or other
vehicle which bounds the use and disclosure of the data.</p>
  </li>
  <li>
    <p>Investigator(s) will by default share de-identified data.</p>
  </li>
  <li>
    <p>Investigator(s) will strive to harmonization of data collection and archiving methods (storage) tools and representation to ensure validation of scientific quality.</p>
  </li>
  <li>
    <p>Investigator(s) will adhere to computer security practices that ensure only authorized individuals can gain access to data files, minimizes unintended access and otherwise adhere to institutional security requirements.</p>
  </li>
</ul>

<h3 id="nih-genomic-data-sharing-gds-policy">NIH Genomic Data Sharing (GDS) Policy</h3>

<p>The GDS Policy expects that large-scale genomic research data from
NIH-supported studies involving human specimens will be submitted to an
NIH-designated data repository. Non-human data may be submitted to any
widely used repository or to the same repositories they submitted
specific types of data to previously. NIH has provided examples of
relevant databases on the NIH Office of Science Policy website.</p>

<ul>
  <li>
    <p>Investigators should submit large-scale human genomic data as well as relevant associated data (e.g., phenotype and exposure data) to an NIH-designated data repository in a timely manner.</p>
  </li>
  <li>
    <p>Investigators should also submit any information necessary to interpret the submitted genomic data, such as study protocols, data instruments, and survey tools.</p>
  </li>
  <li>
    <p>Grants applications and protocols should include details of the required data sharing plan.</p>
  </li>
  <li>
    <p>In general, consent documents should include language that allows for the broad future sharing of genomic data. The NIH recognizes that there will be instances where broad sharing may not be appropriate and the policy outlines the exceptions.</p>
  </li>
  <li>
    <p>Under the GDS Policy, the release of human data for secondary research can generally be deferred for up to six months after data submission, with no publication embargo upon data release.</p>
  </li>
  <li>
    <p>An IRB must review the foundational consents (all versions) to confirm that consent was appropriately sought, includes broad sharing (if appropriate), and to identify any limitations on the upload.</p>
  </li>
</ul>

<p>NOTE: If you are generating genomic data that does not meet the
definition of “large-scale,” consider including a section in your
Resource Sharing Plan that says, “Genomic Data Sharing: Not Applicable”
with a brief explanation. Grant applications are not always sufficiently
detailed for staff to determine whether the GDS policy applies, in which
case staff must contact the PI, which can cause delays.</p>

<ul>
  <li>
    <p>The NIH Genomic Data Sharing Policy is <a href="https://osp.od.nih.gov/scientific-sharing/genomic-data-sharing/">here.</a></p>
  </li>
  <li>
    <p>Large-scale genomic data include genome-wide association studies (GWAS), single nucleotide polymorphisms (SNP) arrays, and genome sequence, transcriptomic, metagenomic, epigenomic, and gene expression data. Examples of research that are subject to the GDS Policy include, but are not limited to, projects that involve generating the whole genome sequence data for more than one gene from more than 1,000 individuals, or analyzing 300,000 or more genetic variants in more than 1,000 individuals, or sequencing more than a 100 isolates of infectious organisms such as bacteria. The <a href="https://osp.od.nih.gov/wp-content/uploads/Supplemental_Info_GDS_Policy.pdf">Supplemental Information to the NIH Genomic Data Sharing Policy</a> includes detailed description of research under scope of the policy and data submission expectations.</p>
  </li>
</ul>

<h3 id="nih-public-genomic-data-repositories">NIH Public Genomic Data Repositories</h3>

<p>NIH is committed to respecting the privacy and intentions of research
participants with regard to how data pertaining to their individual
information is used. Data access is therefore intended only for
scientific investigators pursuing research questions that are consistent
with the informed consent agreements provided by individual research
participants. Furthermore, investigators provided access will be
expected to utilize appropriate controls and abide by Data Use
Limitation.</p>

<p>The dbGAP has two repositories into which researchers can deposit or
withdraw de-identified data:</p>

<ul>
  <li>
    <p>Public Access: Non-individual genomic data can be publicly accessed through the dbGAP website.</p>
  </li>
  <li>
    <p>Controlled Access (Individual-level data): Individual-level data submitted to the dbGaP is required to be de-identified. No names or identifiable information is attached to the data. The genetic fingerprint however is embedded in individual’s genotype data, which is not de-identifiable. That is why, to protect individuals privacy, all individual level data is only distributed through the NIH Authorized Access System.</p>
  </li>
</ul>

<p>Additional information regarding the process of depositing data into NIH Public Repositories can be found from <a href="https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetPdf.cgi?document_name=HowToSubmit.pdf">NCBI</a> and from the <a href="https://datascience.cancer.gov/data-sharing/submitting-data#expectations">NCI</a>.</p>

<h2 id="some-public-repositories">Some Public Repositories</h2>

<p>We here provide some details on a few (by no mean exhaustive) public repositories. If you know of another repository that would like to see added to this page please contact us. Some of these repositories do require IRB approval to download the data. Please correspond with the proper administration authority.</p>

<ul>
  <li><a href="https://gtexportal.org/home/">GTEx</a>: The Genotype-Tissue Expression  project is aiming to understand the difference and regulation of gene expression by tissue across 53 different tissues.</li>
  <li><a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga">TCGA</a>: The Cancer Genome Atlas is a joint effort by the NCI and the NHGRI to moleculary characterize approximately 20000 primary cancer in comparison to matched normal over 33 cancer types.</li>
  <li><a href="http://www.internationalgenome.org/home">1000 genomes</a>: The 1000 genomes project sequenced 1000 genomes to find genetic variants with at least 1% variation in the human population. A useful source</li>
  <li><a href="https://www.ncbi.nlm.nih.gov/geo/">GEO</a>: Gene expression Omnbinus  is a public genomics data repository.</li>
  <li><a href="https://gnomad.broadinstitute.org">gnomAD</a>: The genotype aggression database  is working to aggregate and harmonize sequencing data across multiple studies.</li>
  <li><a href="https://www.nhlbiwgs.org">TOPMed</a>: TOPMed is brought to you by NIH and NHLBI. Trans-Omics for Precision Medicine (TOPmed) aims to “understand of the fundamental biological processes that underlie heart, lung, blood, and sleep (HLBS) disorders.”</li>
  <li><a href="https://www.ncbi.nlm.nih.gov/snp">dbsnp</a>: Provides information about single nuclotide polymorphisms (SNPs), as well as tracking these SNPs publication history.</li>
  <li><a href="https://www.ukbiobank.ac.uk">UK Biobank</a>: Prospective cohort study with a wealth of genetic and health data on 500,000 participants.</li>
  <li><a href="http://www.cbioportal.org/">cBioPortal</a>:  This is an excellent resource for viewing/accessing TCGA cancer genomics data.</li>
  <li><a href="https://www.synapse.org/">Sage’s Synapse.org</a> is not only useful for sharing (privately or publicly) your own research data, but also where <a href="https://www.synapse.org/#!StandaloneWiki:OpenResearchProjects">several open datasets are hosted</a>, and also interact with the various <a href="http://dreamchallenges.org/">DREAM Challenges they host</a>.</li>
</ul>

        
      </section>

      <footer class="page__meta">
        
        


        
          <p class="page__date"><strong><i class="fas fa-fw fa-calendar-alt" aria-hidden="true"></i> Updated:</strong> <time datetime="2019-04-01">April 01, 2019</time></p>
        
      </footer>

      <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_generation/human_shareDeposits.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fgeneration%2Fhuman_shareDeposits%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->


      
  <nav class="pagination">
    
      <a href="/generation/human_privacySecurity/" class="pagination--pager" title="Data Privacy and Security
">Previous</a>
    
    
      <a href="/generation/labman_overview/" class="pagination--pager" title="Overview of Laboratory Management Resources
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form onsubmit="return googleCustomSearchExecute();" id="cse-search-box-form-id">
    <input type="text" id="cse-search-input-box-id" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
    </form>
    <div id="results" class="results">
        <gcse:searchresults-only></gcse:searchresults-only>    
    </div></div>

      </div>
    

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
          <li><a href="/contributors/"><i class="fas fa-user-edit"></i> See our Contributors</a></li>

          <li><a href="https://github.com/FredHutch/wiki"><i class="fab fa-fw fa-github"></i> See our GitHub Repo</a></li>

          <li><a href="https://fhbig.slack.com/"><i class="fab fa-fw fa-slack"> </i> Join the Conversation on Slack</a></li>

          <li><a href="https://fredhutch.github.io/FHBig/"><img src="/images/FHBig.png" alt="fhbig" width="40" align="left|middle" hspace="0"> Find Upcoming Events</a></li>

          <li><a href="https://www.fredhutch.io/"><img src="/images/fhio.png" alt="fredhutchio" width="20" align="left|middle" hspace="0"> Find Training</a></li>


  </ul>
</div>

<div class="page__footer-copyright">&copy; 2019 Fred Hutch Biomedical Data Science Wiki. Powered by <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>
  <script defer src="https://use.fontawesome.com/releases/v5.7.1/js/all.js" integrity="sha384-eVEQC9zshBn0rFj4+TU78eNA19HMNigMviK/PU/FFjLXqa/GKPgX58rvt5Z8PLs7" crossorigin="anonymous"></script>


<script>
  (function () {
    var cx = '018020560330782233421:waz3ch9w_wg';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();

  function googleCustomSearchExecute() {
    var input = document.getElementById('cse-search-input-box-id');
    var element = google.search.cse.element.getElement('searchresults-only0');
    if (input.value == '') {
      element.clearAllResults();
    } else {
      element.execute(input.value);
    }
    return false;
  }

  
</script>





  </body>
</html>
