<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.15.2 by Michael Rose
  Copyright 2013-2019 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Genomics Platforms and Data Types - Fred Hutch Biomedical Data Science Wiki</title>
<meta name="description" content="This guide highlights some of the genomics platforms available through the Genomics Shared Resource at Fred Hutch.  This guide is intended to give general context to each platform.  Access to many of the submission processes involved in using the Genomics Shared Resource is via Hutchbase.">



<meta property="og:type" content="article">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Fred Hutch Biomedical Data Science Wiki">
<meta property="og:title" content="Genomics Platforms and Data Types">
<meta property="og:url" content="http://localhost:4000/generation/datagen_platformsData/">


  <meta property="og:description" content="This guide highlights some of the genomics platforms available through the Genomics Shared Resource at Fred Hutch.  This guide is intended to give general context to each platform.  Access to many of the submission processes involved in using the Genomics Shared Resource is via Hutchbase.">







  <meta property="article:published_time" content="2019-04-02T18:09:25-07:00">



  <meta property="article:modified_time" content="2018-10-25T00:00:00-07:00">



  

  


<link rel="canonical" href="http://localhost:4000/generation/datagen_platformsData/">







  <script type="application/ld+json">
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000",
      "sameAs": null
    }
  </script>







<!-- end _includes/seo.html -->


<link href="index.html" type="application/atom+xml" rel="alternate" title="Fred Hutch Biomedical Data Science Wiki Feed">

<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">

<!--[if IE ]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->



    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
              <a href="http://localhost:4000"><img src="/images/Biomedical-Data-Science-Wiki-logo.png" alt="FH biomedical data science wiki icon" width="600" align="left|middle" hspace="0"></a>
        <!--<a class="site-title" href="/">Fred Hutch Biomedical Data Science Wiki</a>-->

        <ul class="visible-links">
          
          
          <li class="masthead__menu-item">
            <a href="/generation/gen_index/" >Data Generation</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/bioinformatics/inf_index/" >Bioinformatics</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/computing/comp_index/" >Computing</a>
          </li>
          
        </ul>
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z"
              transform="translate(-.01)"></path>
          </svg>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">
  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Data Generation</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/gen_index/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generationdemos/" class="">Resource Library</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Study Design and Funding</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/study_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/study_proposal/" class="">Proposals and Funding</a></li>
          
            
            

            
            

            <li><a href="/generation/study_hypothesis/" class="">Hypothesis Development and Big Molecular Data</a></li>
          
            
            

            
            

            <li><a href="/generation/multiple_testing/" class="">Multiple Testing</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Consent, Privacy and Security</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/human_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/human_consentIRB/" class="">Consenting and Big Molecular Data</a></li>
          
            
            

            
            

            <li><a href="/generation/human_privacySecurity/" class="">Data Privacy and Security</a></li>
          
            
            

            
            

            <li><a href="/generation/security_legal/" class="">Compliance and Legal</a></li>
          
            
            

            
            

            <li><a href="/generation/human_deidentification/" class="">De-identification</a></li>
          
            
            

            
            

            <li><a href="/generation/human_shareDeposits/" class="">Data Sharing and Public Repositories</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Clinical and Experimental Data</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/clsp_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_clinicalCov/" class="">Clinical Data</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_specimenBanking/" class="">Specimen Banking</a></li>
          
            
            

            
            

            <li><a href="/generation/clsp_labCov/" class="">Experimental Data</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Large Scale Data Generation</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/datagen_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_platformsData/" class="active">Genomics Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_flowcytometry/" class="">Flow Cytometry Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_imaging/" class="">Imaging Data</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_proteomics/" class="">Proteomics Data</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Other Laboratory Resources</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/generation/labman_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_assayPrep/" class="">Assay Material Prep and QC</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_dnaApproaches/" class="">DNA-Based Approaches</a></li>
          
            
            

            
            

            <li><a href="/generation/datagen_rnaApproaches/" class="">RNA-Based Approaches</a></li>
          
            
            

            
            

            <li><a href="/generation/labman_software/" class="">Support Software</a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>
    
  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="Genomics Platforms and Data Types">
    <meta itemprop="description" content="This guide highlights some of the genomics platforms available through the Genomics Shared Resource at Fred Hutch.  This guide is intended to give general context to each platform.  Access to many of the submission processes involved in using the Genomics Shared Resource is via Hutchbase.">
    <meta itemprop="datePublished" content="April 02, 2019">
    <meta itemprop="dateModified" content="October 25, 2018">

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Genomics Platforms and Data Types
</h1>
          
          <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_generation/datagen_platformsData.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fgeneration%2Fdatagen_platformsData%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->

        </header>
      


      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right ">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On This Page</h4></header>
              <ul class="toc__menu">
  <li><a href="#sequencing-based-platforms">Sequencing Based Platforms</a>
    <ul>
      <li><a href="#illumina-sequencers">Illumina Sequencers</a></li>
      <li><a href="#pacific-biosciences-pacbio-long-range-sequencer">Pacific Biosciences (PacBio) Long Range Sequencer</a></li>
    </ul>
  </li>
  <li><a href="#array-based-platforms">Array Based Platforms</a>
    <ul>
      <li><a href="#single-nucleotide-polymorphism-snp-arrays-or-methylation-arrays">Single Nucleotide Polymorphism (SNP) Arrays or Methylation Arrays</a></li>
      <li><a href="#nanostring-hybridization-arrays-for-gene-expression">Nanostring Hybridization Arrays for Gene Expression</a></li>
    </ul>
  </li>
  <li><a href="#library-preparation-reagents-and-methods">Library Preparation Reagents and Methods</a>
    <ul>
      <li><a href="#library-preparation-for-sequencing">Library Preparation for Sequencing</a></li>
      <li><a href="#10x-genomics-single-cell-library-preparation-system">10x Genomics Single Cell Library Preparation System</a></li>
      <li><a href="#overview-of-the-sequencing-process">Overview of the Sequencing Process</a></li>
      <li><a href="#available-resources">Available Resources</a></li>
    </ul>
  </li>
</ul>
            </nav>
          </aside>
        
        <p>This guide highlights some of the genomics platforms available through the Genomics Shared Resource at Fred Hutch.  This guide is intended to give general context to each platform.  Access to many of the submission processes involved in using the Genomics Shared Resource is via <a href="https://hutchbase.fhcrc.org/base2/">Hutchbase.</a></p>

<blockquote>
  <p>Note:  As technologies and reagents change, the relative costs of performing these experiments do as well.  It is important to discuss your particular experiment and needs with the Genomics and Bioinformatics Shared Resource during the planning stages of your project.  You can contact them by emailing <code class="highlighter-rouge">genomics</code>.</p>
</blockquote>

<h2 id="sequencing-based-platforms">Sequencing Based Platforms</h2>
<p>Sequencing based platforms currently available via the Genomics Core at the Fred Hutch include the Illumina sequencers (HiSeq/MiSeq) and the Pacific Biosciences Long Range sequencer.  These platforms can be used to sequence a variety of assay material types via different library preparation processes. In the RNA or DNA Approaches pages, we discuss different options for the creation of libraries for sequencing from either nucleic acid type and for different research questions.  Choosing the appropriate assay material QC and library preparation reagents depends in part on how the libraries will be sequenced.  Thus is it important to verify that all phases of the process are using compatible techniques.</p>

<h3 id="illumina-sequencers">Illumina Sequencers</h3>
<p>The Illumina sequencers are high read number, short read sequencers that provide a range of sequencing capabilities for many different upstream library types. The primary approach of these sequencers is to sequence at high numbers, individual fragments of DNA generated by the library preparation process, then to reconstruct the sequences in the mixture by, for example, aligning the sequences of the short reads to a reference, or other bioinformatic approach. The Illumina HiSeq at the Fred Hutch currently has two options:  the High Output mode or the Rapid Run mode.  The basic details for each mode along with the corresponding information for the MiSeq are included below.  We have provided cost ranges to provide a basic idea of the relative costs of these sequencing platforms, but the exact costs of a sequencing run will depend on the read lengths, whether the sequencing is paired end or single end, as well as your affiliation (Fred Hutch vs external).</p>

<table>
  <thead>
    <tr>
      <th>Sequencer</th>
      <th>Mode</th>
      <th>Read Lengths</th>
      <th>Approx. Reads per Lane</th>
      <th>Lanes per Run</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>NovaSeq 6000</td>
      <td>Depends on Chip Type</td>
      <td>50-150, depending on Chip Type</td>
      <td>Variable</td>
      <td>Variable</td>
    </tr>
    <tr>
      <td>HiSeq 2500</td>
      <td>High Output</td>
      <td>50, 100, 125</td>
      <td>250M</td>
      <td>8*</td>
    </tr>
    <tr>
      <td>HiSeq 2500</td>
      <td>Rapid Run</td>
      <td>50, 100, 150, 200, 250</td>
      <td>150M</td>
      <td>2</td>
    </tr>
    <tr>
      <td>MiSeq</td>
      <td>Version 2 Reagents</td>
      <td>up to 250</td>
      <td>15M</td>
      <td>1</td>
    </tr>
    <tr>
      <td>MiSeq</td>
      <td>Version 3 Reagents</td>
      <td>up to 300</td>
      <td>25M</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

<p>* High Output mode requires all 8 lanes to be run simultaneously, no partial runs.</p>

<p>When deciding how much sequencing is needed for a set of libraries to provide sufficient read depth (number of reads per genomic location in the genome covered in the library), issues such as the intended data type, sample type and quality, library preparation type, number of total samples, and the applicability of multiplexing approaches need to be considered.  Consulting with the Genomics Core can help provide more clarity for individual projects.</p>

<p><a href="https://www.illumina.com/techniques/sequencing.html">More from Illumina about Illumina Sequencing can be found here.</a></p>

<h3 id="pacific-biosciences-pacbio-long-range-sequencer">Pacific Biosciences (PacBio) Long Range Sequencer</h3>
<p>The PacBio SMRT sequencer works differently than the Illumina sequencers in that the read length is not specified by the platform, but is limited by the library itself, with an associated reduction in confidence of the sequence as reads get longer and longer.  However, instead of being limited to sequencing only fragments of DNA, PacBio sequencing can provide long stretches of sequencing data that occur in the same fragment.  This allow for analyses such as full length isoform discovery, de novo small genome sequencing, assessing structural variants/translocations, and allele phasing.  On average the PacBio sequencer aims to provide up to 15kb of read length.</p>

<ul>
  <li>300-600k reads per SMRT cell</li>
  <li>Insert size for sequencing of 200bp minimum up to 40kb fragments</li>
  <li>Library prep
    <ul>
      <li>Reagents purchased by lab from PacBio and prepped by lab, brought to Genomics as completed library (see PacBio website for more info)</li>
      <li>OR</li>
      <li>Genomics Shared Resources can help with library prep for a service fee to process a QC’d sample into a Pac Bio library depending on library type:
        <ul>
          <li>For amplicon library prep up to 5kb</li>
          <li>For large insert library prep up to 20kb</li>
        </ul>
      </li>
    </ul>
  </li>
  <li>Multiple multiplexing schemas (in-line or ligated) - discuss details with Genomics Shared Resource to plan the approach (email <code class="highlighter-rouge">genomics</code>).</li>
</ul>

<p>More about <a href="http://www.pacb.com/smrt-science/smrt-sequencing/">PacBio SMRT Sequencing.</a></p>

<h2 id="array-based-platforms">Array Based Platforms</h2>
<p>Microarrays are a sometimes less costly option that can in some cases be substituted for a wide variety of sequencing types; for example, there are SNP, gene expression, and whole exome arrays. While microarrays are not useful for discovery of novel targets, for well-established targets, assay chemistries and data analysis pipelines are well-vetted.  A discussion with the Genomics Core can be useful in helping you decide the best technologies for your work.</p>

<h3 id="single-nucleotide-polymorphism-snp-arrays-or-methylation-arrays">Single Nucleotide Polymorphism (SNP) Arrays or Methylation Arrays</h3>
<p>The Genomics lab is equipped to run all Illumina genotyping and methylation beadchips (<a href="https://www.illumina.com/techniques/microarrays.html">Illumina Microarrays</a>).  Beadchip kits are sold in a variety of sample size kits and it is important to plan total sample sizes as well as randomizing batches to minimize batch-specific bias. Beadchip kits need to be purchased by the investigator and should be drop-shipped to the Genomics lab.  The Genomics lab then charges a fee for processing which covers all non-Illumina supplies, reagents, and labor.  Investigators should plan to provide genomic DNA for genotyping arrays, which has been quantified by a dsDNA specific method such as picogreen.  For methylation arrays, investigators may submit either genomic DNA or bisulfite converted DNA.  Again, the starting DNA should be quantified by picogreen.</p>

<p>Genotyping BeadChip kits vary in price ($50/sample - $600/sample), depending on content.  The Genomics Shared Resource can process samples to run for a service fee, regardless of the beadchip kit with an additional fee for DNA samples that require bisulfite conversion for methylation arrays.</p>

<h3 id="nanostring-hybridization-arrays-for-gene-expression">Nanostring Hybridization Arrays for Gene Expression</h3>
<p>Reagents for <a href="https://www.nanostring.com">Nanostring</a> arrays can be purchased from Nanostring and total RNA ready to be run can be brought to the Genomics Shared Resource for processing.</p>

<h2 id="library-preparation-reagents-and-methods">Library Preparation Reagents and Methods</h2>
<p>The choice of genomics platform will dictate the needs of the library preparation method.  It is important to understand how library preparation can impact the final data, for example if biases towards detection of specific types of nucleic acids are introduced.  Meeting with an NGS specialist at the Genomics Core will help to guide your path.</p>

<h3 id="library-preparation-for-sequencing">Library Preparation for Sequencing</h3>
<p>The four main steps in preparing RNA or DNA for NGS analysis are:</p>

<ol>
  <li>fragmenting and/or sizing the target sequences to a desired length (via physical, enzymatic, and chemical methods)</li>
  <li>converting target to double-stranded DNA (if RNA)</li>
  <li>attaching oligonucleotide adapters to the ends of target fragments–these adaptors are multi-purpose. They are used to index/barcode the fragments, and allow the fragments to be attached to the flow cell of the sequencer (where the sequencing of the fragments occurs)</li>
  <li>quantitating the final library product for sequencing</li>
</ol>

<h3 id="10x-genomics-single-cell-library-preparation-system">10x Genomics Single Cell Library Preparation System</h3>
<p>To obtain single cell gene expression data from RNA-seq, the Genomics lab uses the <a href="https://www.10xgenomics.com">10X Genomics</a> Single Cell Expression platform.  Starting with a cell suspension, this process partitions cells into droplets for cDNA library preparation.  After library prep, the droplets are pooled, then sequenced on an Illumina sequencer. Unique molecular identifiers used in the library prep allow the sequencing results to be computationally traced back to individual cells.</p>

<h3 id="overview-of-the-sequencing-process">Overview of the Sequencing Process</h3>
<ol>
  <li>The adapter-ligated DNA library is loaded onto a flowcell.</li>
  <li>The fragments are hybridized to the flow cell surface.</li>
  <li>Each bound fragment is amplified into a cluster. This step is known as bridge amplification.</li>
  <li>Fluorescently-labeled nucleotides and sequencing reagents are added; the flow cell is fluorescently imaged after the incorporation of each nucleotide. The color of the fluorescent dyes identifies which base was incorporated. This cycle of nucleotide incorporation and imaging is repeated n times for a n-reads of sequence.</li>
</ol>

<h3 id="available-resources">Available Resources</h3>
<ul>
  <li><a href="https://sharedresources.fredhutch.org/core-facilities/genomics">The Hutch Genomics Shared Resource:</a>
The genomics core is VERY helpful if you need guidance about reagent and platform choice for your samples.  Email <code class="highlighter-rouge">genomics</code> to discuss your particular project.</li>
  <li>Submission of some types of samples for services to the Hutch Genomics Shared Resource is via <a href="https://hutchbase.fhcrc.org/base2/">Hutchbase.</a></li>
  <li>Genohub has a compendium of library prep kits, organized by NGS application type <a href="https://genohub.com/ngs-library-preparation-kit-guide/">here</a>.</li>
  <li>Illumina has a few interactive methods guides to help you find the most appropriate <a href="https://www.illumina.com/techniques/sequencing/ngs-library-prep.html">library prep reagents</a> and <a href="https://www.illumina.com/science/sequencing-method-explorer.html">sequencing methods</a> to use in your experiments.</li>
  <li>A helpful reference in considering library prep methods is <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4351865/">Ordoukhanian’s 2014 paper</a>.</li>
</ul>

        
      </section>

      <footer class="page__meta">
        
        


        
          <p class="page__date"><strong><i class="fas fa-fw fa-calendar-alt" aria-hidden="true"></i> Updated:</strong> <time datetime="2018-10-25">October 25, 2018</time></p>
        
      </footer>

      <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_generation/datagen_platformsData.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fgeneration%2Fdatagen_platformsData%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->


      
  <nav class="pagination">
    
      <a href="/generation/datagen_overview/" class="pagination--pager" title="Overview of Large Scale Data Generation
">Previous</a>
    
    
      <a href="/generation/datagen_proteomics/" class="pagination--pager" title="Proteomic Data and Methods
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form onsubmit="return googleCustomSearchExecute();" id="cse-search-box-form-id">
    <input type="text" id="cse-search-input-box-id" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
    </form>
    <div id="results" class="results">
        <gcse:searchresults-only></gcse:searchresults-only>    
    </div></div>

      </div>
    

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
          <li><a href="/contributors/"><i class="fas fa-user-edit"></i> See our Contributors</a></li>

          <li><a href="https://github.com/FredHutch/wiki"><i class="fab fa-fw fa-github"></i> See our GitHub Repo</a></li>

          <li><a href="https://fhbig.slack.com/"><i class="fab fa-fw fa-slack"> </i> Join the Conversation on Slack</a></li>

          <li><a href="https://fredhutch.github.io/FHBig/"><img src="/images/FHBig.png" alt="fhbig" width="40" align="left|middle" hspace="0"> Find Upcoming Events</a></li>

          <li><a href="https://www.fredhutch.io/"><img src="/images/fhio.png" alt="fredhutchio" width="20" align="left|middle" hspace="0"> Find Training</a></li>


  </ul>
</div>

<div class="page__footer-copyright">&copy; 2019 Fred Hutch Biomedical Data Science Wiki. Powered by <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>
  <script defer src="https://use.fontawesome.com/releases/v5.7.1/js/all.js" integrity="sha384-eVEQC9zshBn0rFj4+TU78eNA19HMNigMviK/PU/FFjLXqa/GKPgX58rvt5Z8PLs7" crossorigin="anonymous"></script>


<script>
  (function () {
    var cx = '018020560330782233421:waz3ch9w_wg';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();

  function googleCustomSearchExecute() {
    var input = document.getElementById('cse-search-input-box-id');
    var element = google.search.cse.element.getElement('searchresults-only0');
    if (input.value == '') {
      element.clearAllResults();
    } else {
      element.execute(input.value);
    }
    return false;
  }

  
</script>





  </body>
</html>
