<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.15.2 by Michael Rose
  Copyright 2013-2019 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Fred Hutch Computing Platforms for Bioinformatics - Fred Hutch Biomedical Data Science Wiki</title>
<meta name="description" content="Researchers are more and more likely to need to analyze raw data sets using some sort of analysis process before they can be interpreted in the context of the scientific question. Raw data, whether from an array or sequencing for example, are not typically directly interpretable results, thus require some degree of processing. The nature of the processing depends on the data type, the platform with which the data were generated, and the biological question being asked of the data set. How this process is performed depends on the specific project, but support and resources for this process are available at the Fred Hutch for a variety of needs.">



<meta property="og:type" content="article">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Fred Hutch Biomedical Data Science Wiki">
<meta property="og:title" content="Fred Hutch Computing Platforms for Bioinformatics">
<meta property="og:url" content="http://localhost:4000/bioinformatics/compute_resoverview/">


  <meta property="og:description" content="Researchers are more and more likely to need to analyze raw data sets using some sort of analysis process before they can be interpreted in the context of the scientific question. Raw data, whether from an array or sequencing for example, are not typically directly interpretable results, thus require some degree of processing. The nature of the processing depends on the data type, the platform with which the data were generated, and the biological question being asked of the data set. How this process is performed depends on the specific project, but support and resources for this process are available at the Fred Hutch for a variety of needs.">







  <meta property="article:published_time" content="2019-04-02T18:09:25-07:00">



  <meta property="article:modified_time" content="2018-11-08T00:00:00-08:00">



  

  


<link rel="canonical" href="http://localhost:4000/bioinformatics/compute_resoverview/">







  <script type="application/ld+json">
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000",
      "sameAs": null
    }
  </script>







<!-- end _includes/seo.html -->


<link href="index.html" type="application/atom+xml" rel="alternate" title="Fred Hutch Biomedical Data Science Wiki Feed">

<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">

<!--[if IE ]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->



    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
              <a href="http://localhost:4000"><img src="/images/Biomedical-Data-Science-Wiki-logo.png" alt="FH biomedical data science wiki icon" width="600" align="left|middle" hspace="0"></a>
        <!--<a class="site-title" href="/">Fred Hutch Biomedical Data Science Wiki</a>-->

        <ul class="visible-links">
          
          
          <li class="masthead__menu-item">
            <a href="/generation/gen_index/" >Data Generation</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/bioinformatics/inf_index/" >Bioinformatics</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/computing/comp_index/" >Computing</a>
          </li>
          
        </ul>
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z"
              transform="translate(-.01)"></path>
          </svg>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">
  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Bioinformatics</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/inf_index/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/bioinformaticsdemos/" class="">Resource Library</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Best Practices for Research Data Management</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_storage/" class="">Data Storage</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_ingest/" class="">Data Ingestion and Public Datasets</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_scratch/" class="">Using Scratch Storage</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_archive/" class="">Data Archiving</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Programming</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/prog_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/prog_R/" class="">R, RStudio</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/prog_python/" class="">Python</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/prog_unixbash/" class="">Other Languages</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Managing Code and Computing</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/compute_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/compute_resoverview/" class="active">Fred Hutch Compute Resources</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/compute_github/" class="">GitHub, Versioning and Sharing</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/compute_docker/" class="">Docker</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Training and References</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/howto_training/" class="">Training Opportunties</a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>
    
  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="Fred Hutch Computing Platforms for Bioinformatics">
    <meta itemprop="description" content="Researchers are more and more likely to need to analyze raw data sets using some sort of analysis process before they can be interpreted in the context of the scientific question. Raw data, whether from an array or sequencing for example, are not typically directly interpretable results, thus require some degree of processing. The nature of the processing depends on the data type, the platform with which the data were generated, and the biological question being asked of the data set. How this process is performed depends on the specific project, but support and resources for this process are available at the Fred Hutch for a variety of needs.">
    <meta itemprop="datePublished" content="April 02, 2019">
    <meta itemprop="dateModified" content="November 08, 2018">

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Fred Hutch Computing Platforms for Bioinformatics
</h1>
          
          <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_bioinformatics/compute_resoverview.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fbioinformatics%2Fcompute_resoverview%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->

        </header>
      


      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right ">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On This Page</h4></header>
              <ul class="toc__menu">
  <li><a href="#desktop-computing-platforms">Desktop computing platforms</a></li>
  <li><a href="#interactive-computing--easy-access-moderate-capability">Interactive Computing:  Easy Access, Moderate Capability</a></li>
  <li><a href="#interactive-computing-command-line-interface-cli-moderate-to-high-capability">Interactive Computing: Command Line Interface (CLI), Moderate to High Capability</a>
    <ul>
      <li><a href="#meet-rhino">Meet Rhino</a></li>
      <li><a href="#the-nomachine-cluster">The NoMachine Cluster</a></li>
    </ul>
  </li>
  <li><a href="#batch-computing-and-the-cloud">Batch Computing and the Cloud</a></li>
</ul>
            </nav>
          </aside>
        
        <p>Researchers are more and more likely to need to analyze raw data sets using some sort of analysis process before they can be interpreted in the context of the scientific question. Raw data, whether from an array or sequencing for example, are not typically directly interpretable results, thus require some degree of processing. The nature of the processing depends on the data type, the platform with which the data were generated, and the biological question being asked of the data set. How this process is performed depends on the specific project, but support and resources for this process are available at the Fred Hutch for a variety of needs.</p>

<p>The first step in doing this work is often as simple as asking “what computing resource do I need to use for this task?”  This section describes a range of computing resource options available to Fred Hutch researchers, all the way from a single laptop to the Fred Hutch on-premise high performance computing cluster (<code class="highlighter-rouge">gizmo</code>) to various cloud-based options, as well as how to get started using each platform.</p>

<h2 id="desktop-computing-platforms">Desktop computing platforms</h2>

<p>While most bioinformatic analysis of genomics data, for example, will likely require more compute resources than are typically available in a desktop computer, some smaller analysis jobs or training can be done on a desktop or laptop.  For these smaller compute jobs, scientific computing languages <a href="https://www.r-project.org">R</a> or <a href="https://www.python.org">Python</a> may be useful (see <a href="https://fredhutch.github.io/wiki/bioinformatics/inf_index/">the Bioinformatics page for more information about these computing languages).</a> You can obtain assistance in installing a copy of R and RStudio on PCs through the Center IT desktop software resource. The installation of a commonly used Python and R package management system called Conda can be done on either PCs or Macs and more information about Anaconda and Miniconda (two versions of the Python oriented version of Conda package manager) can be found <a href="https://conda.io/docs/glossary.html#anaconda">here.</a></p>

<p><img src="/assets/comp_index/2018-06-28-12-56-21.png" alt="" /></p>

<h2 id="interactive-computing--easy-access-moderate-capability">Interactive Computing:  Easy Access, Moderate Capability</h2>
<p>These systems are provided by the Fred Hutch to serve the community with the intention of:</p>
<ol>
  <li>being immediately accessible by everyone, regardless of skill set,</li>
  <li>allowing for interactive development of code and/or approaches to analyses, and</li>
  <li>being supported by compute resources larger than a typical desktop computer.</li>
</ol>

<p>These systems are best used for situations such as training of new users in development of R or python code or curation of Galaxy-style workflows that need connectivity to our local filesystem in a way not accessible via the <a href="https://usegalaxy.org/">public instance of Galaxy.</a> These systems also provide compute resources that are a step above a typical desktop computer, thus are good systems for slightly more compute-intensive jobs that still require interactivity.  Each is a managed resource with actively managed packages/modules/tools depending on the system, and thus work well for fairly standard platforms for interactive computing that does not require highly specialized software components.  Additionally, these systems may not be appropriate for analyses that require a higher level of data security, please contact FH username <code class="highlighter-rouge">scicomp</code> to inquire.</p>

<table>
  <thead>
    <tr>
      <th>Compute Resource</th>
      <th>Access Interface</th>
      <th>Resource Admin</th>
      <th>Connection to FH Data Storage</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><a href="http://rstudio.fhcrc.org">RStudio Server</a></td>
      <td>web, FH credentials, no setup required</td>
      <td>Center IT</td>
      <td>Native to python, default access to /fh/fast</td>
    </tr>
    <tr>
      <td><a href="http://galaxy.fredhutch.org/">Fred Hutch Galaxy Instance</a></td>
      <td>web, platform-specific credentials</td>
      <td>Matsen Group</td>
      <td>Limited to staged data</td>
    </tr>
    <tr>
      <td><a href="https://jupyterhub.fhcrc.org/">JupyterHub</a></td>
      <td>web, FH credentials, no setup required</td>
      <td>Center IT</td>
      <td>Native to python, default access to /fh/fast</td>
    </tr>
  </tbody>
</table>

<h2 id="interactive-computing-command-line-interface-cli-moderate-to-high-capability">Interactive Computing: Command Line Interface (CLI), Moderate to High Capability</h2>
<p>These systems are provided by the Fred Hutch to serve needs that rise above those that can be met using the above listed platforms.  Often reasons to move to these HPC resources include the need for version controlled, specialized package/module/tool configurations, more compute resources, or rapid access to large data sets in data storage locations not accessible with the required security for the data type by the above systems. In the table below, <code class="highlighter-rouge">gizmo</code> is actually the compute resource that can be accessed via multiple tools, which are also listed below.</p>

<table>
  <thead>
    <tr>
      <th>Compute Resource</th>
      <th>Access Interface</th>
      <th>Resource Admin</th>
      <th>Connection to FH Data Storage</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Gizmo</td>
      <td>Via Rhino or NoMachine hosts (CLI, FH credentials on campus/VPN off campus)</td>
      <td>Scientific Computing</td>
      <td>Direct to all local storage types</td>
    </tr>
    <tr>
      <td> </td>
      <td> </td>
      <td> </td>
      <td> </td>
    </tr>
    <tr>
      <td>Rhino</td>
      <td>CLI, FH credentials on campus/VPN off campus</td>
      <td>Scientific Computing</td>
      <td>Direct to all local storage types</td>
    </tr>
    <tr>
      <td>NoMachine</td>
      <td>NX Client, FH credentials on campus/VPN off campus</td>
      <td>Scientific Computing</td>
      <td>Direct to all local storage types</td>
    </tr>
    <tr>
      <td>Python/Jupyter Notebooks</td>
      <td>Via Rhino (CLI, FH credentials on campus/VPN off campus)</td>
      <td>Scientific Computing</td>
      <td>Direct to all local storage types</td>
    </tr>
    <tr>
      <td>R/R Studio</td>
      <td>Via Rhino (CLI, FH credentials on campus/VPN off campus)</td>
      <td>Scientific Computing</td>
      <td>Direct to all local storage types</td>
    </tr>
  </tbody>
</table>

<h3 id="meet-rhino">Meet <code class="highlighter-rouge">Rhino</code></h3>
<p><code class="highlighter-rouge">Gizmo</code> is actually not a stand alone system; instead, access to the resource is based on the <code class="highlighter-rouge">Rhino</code> platform supported by Center IT.  <code class="highlighter-rouge">Rhino</code>, or more specifically the Rhinos, are three locally managed HPC servers all accessed via the name <code class="highlighter-rouge">rhino</code>. Together, they function as a data and compute hub for a variety of data storage resources and high performance computing (HPC). The specific guidance for the use of each of the approaches to HPC access are slightly different, but will all require the user to learn how to access and interact with <code class="highlighter-rouge">rhino</code>.</p>

<p>Any user interacting with the following systems will be dependent on being proficient with the care and keeping of the Rhinos, and more information about on premise computing resources can be found in the Computing section of this site <a href="/computing/resource_overview/">here</a>.</p>

<p>More information on the topic of ssh configurations for access to <code class="highlighter-rouge">rhino</code> can be found <a href="/compdemos/ssh/">here.</a></p>

<h3 id="the-nomachine-cluster">The NoMachine Cluster</h3>

<p><a href="https://www.nomachine.com/">NoMachine</a> is a software suite that allows you to run a Linux desktop session remotely.  The session runs on the NoMachine server but is displayed on your desktop or laptop using the NoMachine client.  NoMachine (also abbreviated <em>NX</em>) is installed on CIT supported PC desktops and laptops.</p>

<p>NX has the particular advantage of maintaining your session even when you disconnect or lose connectivity.  All that is required is to restart the client and your session will be as you’d last left it.</p>

<p>There are three systems you can use for NX sessions: <em>lynx</em>, <em>manx</em>, and <em>sphinx</em>.  These are not computational systems but rather these hosts are used solely as launch-points for sessions on <code class="highlighter-rouge">gizmo</code> or <code class="highlighter-rouge">rhino</code>. Running computational tools on this system will get you a warning from SciComp.</p>

<h2 id="batch-computing-and-the-cloud">Batch Computing and the Cloud</h2>

<p><em>Batch computing</em> allows you to queue up jobs and have them executed by the batch system, rather than you having to start an interactive session on a high-performance system.  Using the batch system allows you to queue up thousands of jobs- something impractical to impossible when using an interactive session.  There are benefits when you have a smaller volume of jobs as well- interactive jobs are dependent on the shell from which they are launched- if your laptop disconnected, the job would be terminated.</p>

<p>The batch system used at the Hutch is <a href="http://schedmd.com">Slurm</a>.  Slurm provides a set of commands for submitting and managing jobs on the <code class="highlighter-rouge">gizmo</code> and <code class="highlighter-rouge">beagle</code> clusters as well as providing information on the state (success or failure) and metrics (memory and compute usage) of completed jobs.  More information about on batch computing and cloud based computing resources can be found in the Computing section of this site <a href="/computing/cluster_cloudCompute/">here</a>.</p>

<table>
  <thead>
    <tr>
      <th>Compute Resource</th>
      <th>Access Interface</th>
      <th>Resource Admin</th>
      <th>Connection to FH Data Storage</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Gizmo</td>
      <td>Via Rhino or NoMachine hosts (CLI, FH credentials on campus/VPN off campus)</td>
      <td>Scientific Computing</td>
      <td>Direct to all local storage types</td>
    </tr>
    <tr>
      <td>Beagle</td>
      <td>Via Rhino or NoMachine hosts (CLI, FH credentials on campus/VPN off campus)</td>
      <td>Center IT</td>
      <td><em>home</em>, <em>fast</em>, <em>economy</em>, and S3 access. Beagle has its own <em>scratch</em></td>
    </tr>
    <tr>
      <td>AWS Batch</td>
      <td>Via Rhino or NoMachine hosts (CLI, FH credentials on campus/VPN off campus)</td>
      <td>Scientific Computing</td>
      <td>Direct to all local storage types</td>
    </tr>
  </tbody>
</table>

        
      </section>

      <footer class="page__meta">
        
        


        
          <p class="page__date"><strong><i class="fas fa-fw fa-calendar-alt" aria-hidden="true"></i> Updated:</strong> <time datetime="2018-11-08">November 08, 2018</time></p>
        
      </footer>

      <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_bioinformatics/compute_resoverview.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fbioinformatics%2Fcompute_resoverview%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->


      
  <nav class="pagination">
    
      <a href="/bioinformatics/compute_overview/" class="pagination--pager" title="Managing Code and Computing
">Previous</a>
    
    
      <a href="/bioinformatics/dm_archive/" class="pagination--pager" title="Data Archiving Best Practices
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form onsubmit="return googleCustomSearchExecute();" id="cse-search-box-form-id">
    <input type="text" id="cse-search-input-box-id" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
    </form>
    <div id="results" class="results">
        <gcse:searchresults-only></gcse:searchresults-only>    
    </div></div>

      </div>
    

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
          <li><a href="/contributors/"><i class="fas fa-user-edit"></i> See our Contributors</a></li>

          <li><a href="https://github.com/FredHutch/wiki"><i class="fab fa-fw fa-github"></i> See our GitHub Repo</a></li>

          <li><a href="https://fhbig.slack.com/"><i class="fab fa-fw fa-slack"> </i> Join the Conversation on Slack</a></li>

          <li><a href="https://fredhutch.github.io/FHBig/"><img src="/images/FHBig.png" alt="fhbig" width="40" align="left|middle" hspace="0"> Find Upcoming Events</a></li>

          <li><a href="https://www.fredhutch.io/"><img src="/images/fhio.png" alt="fredhutchio" width="20" align="left|middle" hspace="0"> Find Training</a></li>


  </ul>
</div>

<div class="page__footer-copyright">&copy; 2019 Fred Hutch Biomedical Data Science Wiki. Powered by <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>
  <script defer src="https://use.fontawesome.com/releases/v5.7.1/js/all.js" integrity="sha384-eVEQC9zshBn0rFj4+TU78eNA19HMNigMviK/PU/FFjLXqa/GKPgX58rvt5Z8PLs7" crossorigin="anonymous"></script>


<script>
  (function () {
    var cx = '018020560330782233421:waz3ch9w_wg';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();

  function googleCustomSearchExecute() {
    var input = document.getElementById('cse-search-input-box-id');
    var element = google.search.cse.element.getElement('searchresults-only0');
    if (input.value == '') {
      element.clearAllResults();
    } else {
      element.execute(input.value);
    }
    return false;
  }

  
</script>





  </body>
</html>
