<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.15.2 by Michael Rose
  Copyright 2013-2019 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Data Ingestion and Public Datasets - Fred Hutch Biomedical Data Science Wiki</title>
<meta name="description" content="Large-scale research data can come from multiple sources, like one of the Fred Hutch Shared Resources, external vendors, external collaborators or public repositories. Regardless, if processing or analysis is required, your data will need to be accessible via a Fred Hutch managed data storage, compute resource, or workstation. Being selective about what degree public datasets need to be copied to a local storage space can lower your project costs.">



<meta property="og:type" content="article">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Fred Hutch Biomedical Data Science Wiki">
<meta property="og:title" content="Data Ingestion and Public Datasets">
<meta property="og:url" content="http://localhost:4000/bioinformatics/dm_ingest/">


  <meta property="og:description" content="Large-scale research data can come from multiple sources, like one of the Fred Hutch Shared Resources, external vendors, external collaborators or public repositories. Regardless, if processing or analysis is required, your data will need to be accessible via a Fred Hutch managed data storage, compute resource, or workstation. Being selective about what degree public datasets need to be copied to a local storage space can lower your project costs.">







  <meta property="article:published_time" content="2019-04-02T18:09:25-07:00">



  <meta property="article:modified_time" content="2019-03-19T00:00:00-07:00">



  

  


<link rel="canonical" href="http://localhost:4000/bioinformatics/dm_ingest/">







  <script type="application/ld+json">
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000",
      "sameAs": null
    }
  </script>







<!-- end _includes/seo.html -->


<link href="index.html" type="application/atom+xml" rel="alternate" title="Fred Hutch Biomedical Data Science Wiki Feed">

<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">

<!--[if IE ]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->



    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
              <a href="http://localhost:4000"><img src="/images/Biomedical-Data-Science-Wiki-logo.png" alt="FH biomedical data science wiki icon" width="600" align="left|middle" hspace="0"></a>
        <!--<a class="site-title" href="/">Fred Hutch Biomedical Data Science Wiki</a>-->

        <ul class="visible-links">
          
          
          <li class="masthead__menu-item">
            <a href="/generation/gen_index/" >Data Generation</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/bioinformatics/inf_index/" >Bioinformatics</a>
          </li>
          
          
          <li class="masthead__menu-item">
            <a href="/computing/comp_index/" >Computing</a>
          </li>
          
        </ul>
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z"
              transform="translate(-.01)"></path>
          </svg>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">
  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Bioinformatics</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/inf_index/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/bioinformaticsdemos/" class="">Resource Library</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Best Practices for Research Data Management</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_storage/" class="">Data Storage</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_ingest/" class="active">Data Ingestion and Public Datasets</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_scratch/" class="">Using Scratch Storage</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/dm_archive/" class="">Data Archiving</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Programming</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/prog_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/prog_R/" class="">R, RStudio</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/prog_python/" class="">Python</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/prog_unixbash/" class="">Other Languages</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Managing Code and Computing</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/compute_overview/" class="">Overview</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/compute_resoverview/" class="">Fred Hutch Compute Resources</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/compute_github/" class="">GitHub, Versioning and Sharing</a></li>
          
            
            

            
            

            <li><a href="/bioinformatics/compute_docker/" class="">Docker</a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Training and References</span>
        

        
        <ul>
          
            
            

            
            

            <li><a href="/bioinformatics/howto_training/" class="">Training Opportunties</a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>
    
  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="Data Ingestion and Public Datasets">
    <meta itemprop="description" content="Large-scale research data can come from multiple sources, like one of the Fred Hutch Shared Resources, external vendors, external collaborators or public repositories. Regardless, if processing or analysis is required, your data will need to be accessible via a Fred Hutch managed data storage, compute resource, or workstation. Being selective about what degree public datasets need to be copied to a local storage space can lower your project costs.">
    <meta itemprop="datePublished" content="April 02, 2019">
    <meta itemprop="dateModified" content="March 19, 2019">

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Data Ingestion and Public Datasets
</h1>
          
          <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_bioinformatics/dm_ingest.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fbioinformatics%2Fdm_ingest%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->

        </header>
      


      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right ">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On This Page</h4></header>
              <ul class="toc__menu">
  <li><a href="#publicly-available-datasets">Publicly Available Datasets</a></li>
  <li><a href="#ingestion-of-externally-generated-data">Ingestion of Externally Generated Data</a>
    <ul>
      <li><a href="#ftp-to-scratch">FTP to Scratch</a>
        <ul>
          <li><a href="#helpful-wget-options">Helpful wget Options</a></li>
          <li><a href="#skipping-existing-files">#Skipping Existing Files</a>
            <ul>
              <li><a href="#managing-credentials">Managing Credentials</a></li>
            </ul>
          </li>
        </ul>
      </li>
      <li><a href="#sync-to-aws-s3-economy-cloud-storage">Sync to AWS S3 Economy Cloud Storage</a></li>
      <li><a href="#clean-out-scratch">Clean out Scratch</a></li>
    </ul>
  </li>
</ul>
            </nav>
          </aside>
        
        <p>Large-scale research data can come from multiple sources, like one of the Fred Hutch Shared Resources, external vendors, external collaborators or public repositories. Regardless, if processing or analysis is required, your data will need to be accessible via a Fred Hutch managed data storage, compute resource, or workstation. Being selective about what degree public datasets need to be copied to a local storage space can lower your project costs.</p>

<p>Below, we outline some different approaches to ingest and store only the most relevant portions of the data.</p>

<h2 id="publicly-available-datasets">Publicly Available Datasets</h2>
<p>There are multiple sources and tiers of data available publicly.  In order to avoid, for example, a researcher having to pay to host large, raw datasets that are publicly available, there are approaches to accessing and documenting only the minimum required data.  Knowing how best to approach publicly available large scale data sets can both make a study far more productive in a shorter period of time and save resources from being unnecessarily spent on generating new data or storing copies of existing datasets when not required for the particular research they are being used for.  This section will have more to come on this topic.</p>

<ul>
  <li><a href="http://www.cbioportal.org/">cBioPortal</a> is an excellent web-accessible resource to query various publicly available study data from projects such as TCGA or other more specific studies.</li>
  <li><a href="https://www.synapse.org/#!StandaloneWiki:OpenResearchProjects">Sage Bionetwork’s Synapse platform</a> hosts and organizes several open research projects that involve large scale molecular data sets and researchers can follow their documentation to download data through the web or python clients.</li>
  <li><a href="http://immunespace.org/">ImmuneSpace</a> has publicly-accessible RNASeq, HAI, and flow-cytometry data from the Human Immunology Project Consortium</li>
  <li><a href="https://dataspace.cavd.org">CAVD DataSpace</a> has publicly-accessible data from HIV vaccine studies</li>
</ul>

<h2 id="ingestion-of-externally-generated-data">Ingestion of Externally Generated Data</h2>
<p>For data from non-Fred Hutch entities that you would like to transfer to Fred Hutch-managed storage for further analysis, there is a multi-step process that Scientific Computing can assist you with. Large scale biomedical datasets have higher risks of data corruption and transfer interruption. Also, “intermediate” data may also need to be generated during analyses by Fred Hutch investigators. Thus, it is important for you to work with Scientific Computing (email <code class="highlighter-rouge">scicomp</code>) to ensure that the external data are transferred completely, and you have secure and affordable storage on FredHutch systems.</p>

<p>Before ingesting any external data sets, please refer to the <a href="/computing/store_overview/">Overview of Data Storage at Fred Hutch</a> page to begin to identify the final home of the data you are ingesting.  Choosing a location will depend on how large the data are, how often you need to access them, and what type of security concerns might apply to the dataset.  Once you have an idea where the data will need to live at Fred Hutch, you can begin the process of transferring them to Fred Hutch supported data storage systems.</p>

<p>The process is generally:</p>
<ul>
  <li>Fred Hutch user downloads data from an external source to the <a href="/computing/store_scratch/"><em>Scratch</em> service</a> using ftp, scp, ascp, etc or via <a href="/computing/store_collaboration/">collaborative data storage tools.</a></li>
</ul>

<p>OR</p>

<ul>
  <li>Provide the sequencing center or data source the information needed to copy the data into one of the Fred Hutch Managed <a href="/computing/store_objectstore/#economy-cloud-s3">Amazon S3 transfer buckets</a>.</li>
</ul>

<p>THEN</p>
<ul>
  <li>
    <p>Then validate the md5 checksums of the data against the checksum info (usually a text file containing md5sums) provided by the sequencing center or data source. (This checks for data corruption or incomplete transfers)</p>
  </li>
  <li>
    <p>Transfer the validated data to the PI’s <a href="/computing/store_objectstore/"><em>Economy</em> storage.</a></p>
  </li>
</ul>

<p>Ideally, one wouldn’t transfer large amounts of data intended to be temporary into systems like <em>Fast</em> storage as that data will be included in the frequent snapshots for backup purposes in that storage system.  This means these data will have a large footprint in the overall system, requiring long term backup of what are actually temporary data (ironic in a way). A better way to ingest and store large data sets is to first ingest the data into <em>Scratch</em> storage and then move them into secure, stable object storage systems like <em>Economy</em> (either <em>Local</em> or <em>Cloud</em>).</p>

<h3 id="ftp-to-scratch">FTP to <em>Scratch</em></h3>
<p>First, make a directory in Scratch for the data you want to ingest.  Then use <code class="highlighter-rouge">wget</code> to copy, in this case recursively, the data from the ftp url (e.g., ftp.broadinstitute.org/bundle/) using the username (user) and password (pass) required.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /fh/scratch/delete10/lastname_f/
mkdir ingestedDataDir
cd ingestedDataDir
wget --recursive ftp://user@ftp.broadinstitute.org/bundle/
</code></pre></div></div>

<p>While <code class="highlighter-rouge">wget</code> supports sending a password in the command line (either by using <code class="highlighter-rouge">user:pass@</code> or via the <code class="highlighter-rouge">--password</code> option) this will make your credentials visible to anyone on the system.</p>

<h4 id="helpful-wget-options">Helpful <code class="highlighter-rouge">wget</code> Options</h4>

<h4 id="skipping-existing-files">#Skipping Existing Files</h4>

<p>If you need to repeat the ftp if not all files come down, use no-clobber when resending the command and this will skip any files that already exist.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>--no-clobber
</code></pre></div></div>

<h5 id="managing-credentials">Managing Credentials</h5>

<p>As mentioned above, there are different options for providing credentials (username and password) to the remote site.  The different options expose, to varying degrees, those credentials to others on the system.  Thus it is up to you to determine which of these methods should be used for storage of the credential.</p>

<blockquote>
  <p>IMPORTANT: Passwords are classified as level III data by the ISO.  Thus, none of these mechanisms can be used to store your HutchNet ID.</p>
</blockquote>

<p>The first option is to send the credential on the command line, either in the URI or as an option to <code class="highlighter-rouge">wget</code>.  Providing the credential in the URI means adding the password after the username:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>wget ftp://basil:VeryStrongPassword@site.example.org
</code></pre></div></div>

<p>You can also put the username and password in options for the <code class="highlighter-rouge">wget</code> command:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>wget --user=basil --password=VeryStrongPassword ftp://site.example.org
</code></pre></div></div>

<p>This will expose the username and password to any other users on the system. For anonymous or commonly known credentials this may not be a concern but should not be used for any other credentials.</p>

<p>The next methods allow you to hide the credentials in a file- these files are unencrypted which makes storage of very sensitive credentials inadvisable (or in the case of your HutchNet ID, disallowed). It is <em>essential</em> that this file’s permissions allow read and write by you alone.</p>

<p>You can store these credentials in a file in your home directory.  On startup, <code class="highlighter-rouge">wget</code> will read the file <code class="highlighter-rouge">$HOME/.wgetrc</code>- this file can be used to store credentials:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>user=basil
password=VeryStrongPassword
</code></pre></div></div>

<p>Another way to store credentials is to create a file with the URIs you wish to download.  These URIs would contain the credentials using the <code class="highlighter-rouge">user:pass</code> syntax.  For example, if we create the file <code class="highlighter-rouge">$HOME/data_download.txt</code> with the line:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ftp://basil:VeryStrongPassword@site.example.org
</code></pre></div></div>

<p>You then provide this file to <code class="highlighter-rouge">wget</code> with the following command:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cat $HOME/data_download | wget -i -
</code></pre></div></div>

<p>As indicated above, setting this file to be readable by <em>only</em> you is essential:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>chmod u=rw,go-rwx $HOME/.wgetrc
chmod u=rw,go-rwx $HOME/data_download.txt
</code></pre></div></div>

<h3 id="sync-to-aws-s3-economy-cloud-storage">Sync to AWS S3 <em>Economy Cloud</em> Storage</h3>
<p>Once you’ve confirmed that the intended data has been fully transferred to <em>Scratch</em> then transfer these data to the intended final location in <em>Economy</em>, in this case AWS S3 (<em>Economy Cloud</em>).</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>aws s3 sync /ingestedDataDir/ s3://your-pi-bucket-name/theseData/
</code></pre></div></div>

<h3 id="clean-out-scratch">Clean out <em>Scratch</em></h3>
<p>While the data in <em>Scratch</em> are not backed up and regularly deleted, the system does have a finite amount of space available to users.  Especially when your temporary data is 100s of GB or into the TB range, it’s preferred that after the data are confirmed to have been transferred to the final location that you remove your own data in <em>Scratch</em>.  This eliminates the possibility of someone else having to delete your data if another investigator needs a large amount of space.</p>

<blockquote>
  <p>Note:  Only ever use <code class="highlighter-rouge">rm -rf</code> when you are CERTAIN that a complete, permanent delete that cannot be retrieved is what you want.</p>
</blockquote>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>rm -rf /fh/scratch/delete10/lastname_f/ingestedDataDir
</code></pre></div></div>

        
      </section>

      <footer class="page__meta">
        
        


        
          <p class="page__date"><strong><i class="fas fa-fw fa-calendar-alt" aria-hidden="true"></i> Updated:</strong> <time datetime="2019-03-19">March 19, 2019</time></p>
        
      </footer>

      <!-- <section class="page__share"> -->
<font size="2">
  <a href="http://github.com/FredHutch/wiki/blob/master/_bioinformatics/dm_ingest.md"><i class="fas fa-pen-square"></i>
  <span> Edit this Page via GitHub</span></a> &nbsp; &nbsp; &nbsp;
  <a href="http://github.com/FredHutch/wiki/issues"><i class="far fa-comment-dots"></i><span> Comment by Filing an Issue</span></a>&nbsp; &nbsp; &nbsp;
  <a href="https://fhbig.slack.com/#question-and-answer"><i class="fas fa-question-circle"></i>
  <span> Have Questions? Ask them here.</span></a>
</font>
<hr>

<!--  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A4000%2Fbioinformatics%2Fdm_ingest%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a> -->
<!-- </section> -->


      
  <nav class="pagination">
    
      <a href="/bioinformatics/dm_archive/" class="pagination--pager" title="Data Archiving Best Practices
">Previous</a>
    
    
      <a href="/bioinformatics/dm_overview/" class="pagination--pager" title="Overview of Data Management for Researchers
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form onsubmit="return googleCustomSearchExecute();" id="cse-search-box-form-id">
    <input type="text" id="cse-search-input-box-id" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
    </form>
    <div id="results" class="results">
        <gcse:searchresults-only></gcse:searchresults-only>    
    </div></div>

      </div>
    

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
          <li><a href="/contributors/"><i class="fas fa-user-edit"></i> See our Contributors</a></li>

          <li><a href="https://github.com/FredHutch/wiki"><i class="fab fa-fw fa-github"></i> See our GitHub Repo</a></li>

          <li><a href="https://fhbig.slack.com/"><i class="fab fa-fw fa-slack"> </i> Join the Conversation on Slack</a></li>

          <li><a href="https://fredhutch.github.io/FHBig/"><img src="/images/FHBig.png" alt="fhbig" width="40" align="left|middle" hspace="0"> Find Upcoming Events</a></li>

          <li><a href="https://www.fredhutch.io/"><img src="/images/fhio.png" alt="fredhutchio" width="20" align="left|middle" hspace="0"> Find Training</a></li>


  </ul>
</div>

<div class="page__footer-copyright">&copy; 2019 Fred Hutch Biomedical Data Science Wiki. Powered by <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>
  <script defer src="https://use.fontawesome.com/releases/v5.7.1/js/all.js" integrity="sha384-eVEQC9zshBn0rFj4+TU78eNA19HMNigMviK/PU/FFjLXqa/GKPgX58rvt5Z8PLs7" crossorigin="anonymous"></script>


<script>
  (function () {
    var cx = '018020560330782233421:waz3ch9w_wg';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();

  function googleCustomSearchExecute() {
    var input = document.getElementById('cse-search-input-box-id');
    var element = google.search.cse.element.getElement('searchresults-only0');
    if (input.value == '') {
      element.clearAllResults();
    } else {
      element.execute(input.value);
    }
    return false;
  }

  
</script>





  </body>
</html>
