---
title: Nextflow Workflows Catalog
---

This table provides a collection of bioinformatics workflows designed for a variety of applications in genomic and transcriptomic research. Each workflow is linked to its corresponding GitHub repository, where you can find documentation on how to run the analysis, as well as instructions for filing an issue if you need assistance. The workflows cover a wide range of tasks, including RNA sequencing analysis, microbial genome assembly, quality control, and more.

Use the descriptions provided to determine which workflow best suits your research needs. If you're looking for a specific analysis type, such as differential expression analysis, microbial taxonomic classification, or gene set enrichment, simply look for the relevant workflow and follow the link to the GitHub repository for detailed usage instructions.

| Workflow | Purpose | Homepage |
|----------|---------|----------|
| **RNAseq - Gene Expression Analysis** | Quantify the degree of gene expression from RNA sequencing datasets | [GitHub](https://github.com/nf-core/rnaseq) |
| **Differential Expression Analysis** | Identify the genes that are differentially expressed between treatment and control groups | [GitHub](https://github.com/FredHutch/pw-differential-expression) |
| **Gene Set Enrichment Analysis** | Identify the gene sets that are enriched in a collection of differentially expressed genes | [GitHub](https://github.com/FredHutch/pw-gene-set-enrichment) |
| **Bulk RNA Immune Clonotypes** | Identify the immune clonotypes present from bulk RNA sequencing data | [GitHub](https://github.com/FredHutch/pw-bulk-rna-immune-clonotypes) |
| **Bulk RNA Lymphocyte Abundance Estimation** | Estimate the proportion of lymphocyte cells from bulk RNA gene expression data | [GitHub](https://github.com/FredHutch/pw-bulk-rna-lymphocytes) |
| **Microbial Taxonomic Classification (metaphlan2)** | Estimate the abundance of microbial taxa from WGS data | [GitHub](https://github.com/FredHutch/metaphlan-nf) |
| **CRISPR Screen Analysis** | Identify genes that are enriched/depleted from CRISPR screen experiments | [GitHub](https://github.com/FredHutch/crispr-screen-nf) |
| **PacBio CCS Analysis** | Extract CCS from CLR data generated by the PacBio Sequel instrument | [GitHub](https://github.com/FredHutch/ccs-nf) |
| **CellProfiler Batch Analysis** | Process images in parallel through a CellProfiler pipeline and combine the results | [GitHub](https://github.com/FredHutch/cellprofiler-batch-nf) |
| **Gene-Level Metagenomic Analysis (geneshot)** | Perform gene-level metagenomic analysis of WGS datasets for strain-level microbiome analysis | [GitHub](https://github.com/Golob-Minot/geneshot) |
| **FASTQ Quality Assessment** | Generate a QC summary over a collection of FASTQ input files | [GitHub](https://github.com/FredHutch/multi-fastqc-nf) |
| **Microbial Operon Identification** | Identify genomes that contain a set of genes in close proximity | [GitHub](https://github.com/FredHutch/octapus) |
| **Microbial Pan-Genome Analysis (anvi'o)** | Analyze a set of microbial genomes with the [anvi'o](https://github.com/merenlab/anvio) pangenome pipeline | [GitHub](https://github.com/FredHutch/nf-anvio-pangenome) |
| **Microbial Genome Annotation (PGAP)** | Annotate a set of bacterial genomes using the [PGAP annotation software suite](https://www.ncbi.nlm.nih.gov/genome/annotation_prok/) | [GitHub](https://github.com/FredHutch/PGAP-nf) |
| **Microbial Genome Annotation (Prokka)** | Annotate a set of bacterial genomes using the [Prokka annotation software suite](https://github.com/tseemann/prokka) | [GitHub](https://github.com/FredHutch/prokka-nf) |
| **Microbial Genome Assembly (UniCycler)** | Perform _de novo_ assembly of bacterial genomes using the [UniCycler](https://github.com/rrwick/Unicycler) hybrid assembler | [GitHub](https://github.com/FredHutch/unicycler-nf) |



## Workflow Support
If any of these materials are insufficient you can get support via:
- The FH Data Slack [#workflow-managers](https://fhdata.slack.com/archives/CJFP1NYSZ) channel
- Schedule a [Research Computing Data House Call](https://calendly.com/data-house-calls/computing?back=1&month=2024-04) 


